// @kvib/react/dist/index.js
'use strict';

var react = require('@chakra-ui/react');
var jsxRuntime = require('react/jsx-runtime');
var material_symbols_star = require('material-symbols');
require('@fontsource/mulish/400.css');
require('@fontsource/mulish/600.css');
require('@fontsource/mulish/700.css');
var react$1 = require('react');
var skipNav = require('@chakra-ui/skip-nav');
var reactDayPicker = require('react-day-picker');
require('react-day-picker/dist/style.css');
var nb = require('date-fns/locale/nb');
var dateFns = require('date-fns');
var chakraReactSelect = require('chakra-react-select');
var reactStately = require('react-stately');
var reactAria = require('react-aria');

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var material_symbols_star__namespace = /*#__PURE__*/_interopNamespace(material_symbols_star);

var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));

// ../../node_modules/lodash.mergewith/index.js
var require_lodash = __commonJS({
  "../../node_modules/lodash.mergewith/index.js"(exports, module) {
    var LARGE_ARRAY_SIZE = 200;
    var HASH_UNDEFINED = "__lodash_hash_undefined__";
    var HOT_COUNT = 800;
    var HOT_SPAN = 16;
    var MAX_SAFE_INTEGER = 9007199254740991;
    var argsTag = "[object Arguments]";
    var arrayTag = "[object Array]";
    var asyncTag = "[object AsyncFunction]";
    var boolTag = "[object Boolean]";
    var dateTag = "[object Date]";
    var errorTag = "[object Error]";
    var funcTag = "[object Function]";
    var genTag = "[object GeneratorFunction]";
    var mapTag = "[object Map]";
    var numberTag = "[object Number]";
    var nullTag = "[object Null]";
    var objectTag = "[object Object]";
    var proxyTag = "[object Proxy]";
    var regexpTag = "[object RegExp]";
    var setTag = "[object Set]";
    var stringTag = "[object String]";
    var undefinedTag = "[object Undefined]";
    var weakMapTag = "[object WeakMap]";
    var arrayBufferTag = "[object ArrayBuffer]";
    var dataViewTag = "[object DataView]";
    var float32Tag = "[object Float32Array]";
    var float64Tag = "[object Float64Array]";
    var int8Tag = "[object Int8Array]";
    var int16Tag = "[object Int16Array]";
    var int32Tag = "[object Int32Array]";
    var uint8Tag = "[object Uint8Array]";
    var uint8ClampedTag = "[object Uint8ClampedArray]";
    var uint16Tag = "[object Uint16Array]";
    var uint32Tag = "[object Uint32Array]";
    var reRegExpChar = /[\\^$.*+?()[\]{}|]/g;
    var reIsHostCtor = /^\[object .+?Constructor\]$/;
    var reIsUint = /^(?:0|[1-9]\d*)$/;
    var typedArrayTags = {};
    typedArrayTags[float32Tag] = typedArrayTags[float64Tag] = typedArrayTags[int8Tag] = typedArrayTags[int16Tag] = typedArrayTags[int32Tag] = typedArrayTags[uint8Tag] = typedArrayTags[uint8ClampedTag] = typedArrayTags[uint16Tag] = typedArrayTags[uint32Tag] = true;
    typedArrayTags[argsTag] = typedArrayTags[arrayTag] = typedArrayTags[arrayBufferTag] = typedArrayTags[boolTag] = typedArrayTags[dataViewTag] = typedArrayTags[dateTag] = typedArrayTags[errorTag] = typedArrayTags[funcTag] = typedArrayTags[mapTag] = typedArrayTags[numberTag] = typedArrayTags[objectTag] = typedArrayTags[regexpTag] = typedArrayTags[setTag] = typedArrayTags[stringTag] = typedArrayTags[weakMapTag] = false;
    var freeGlobal = typeof global == "object" && global && global.Object === Object && global;
    var freeSelf = typeof self == "object" && self && self.Object === Object && self;
    var root = freeGlobal || freeSelf || Function("return this")();
    var freeExports = typeof exports == "object" && exports && !exports.nodeType && exports;
    var freeModule = freeExports && typeof module == "object" && module && !module.nodeType && module;
    var moduleExports = freeModule && freeModule.exports === freeExports;
    var freeProcess = moduleExports && freeGlobal.process;
    var nodeUtil = function() {
      try {
        var types = freeModule && freeModule.require && freeModule.require("util").types;
        if (types) {
          return types;
        }
        return freeProcess && freeProcess.binding && freeProcess.binding("util");
      } catch (e) {
      }
    }();
    var nodeIsTypedArray = nodeUtil && nodeUtil.isTypedArray;
    function apply(func, thisArg, args) {
      switch (args.length) {
        case 0:
          return func.call(thisArg);
        case 1:
          return func.call(thisArg, args[0]);
        case 2:
          return func.call(thisArg, args[0], args[1]);
        case 3:
          return func.call(thisArg, args[0], args[1], args[2]);
      }
      return func.apply(thisArg, args);
    }
    function baseTimes(n, iteratee) {
      var index = -1, result = Array(n);
      while (++index < n) {
        result[index] = iteratee(index);
      }
      return result;
    }
    function baseUnary(func) {
      return function(value) {
        return func(value);
      };
    }
    function getValue(object, key) {
      return object == null ? void 0 : object[key];
    }
    function overArg(func, transform2) {
      return function(arg) {
        return func(transform2(arg));
      };
    }
    var arrayProto = Array.prototype;
    var funcProto = Function.prototype;
    var objectProto = Object.prototype;
    var coreJsData = root["__core-js_shared__"];
    var funcToString = funcProto.toString;
    var hasOwnProperty = objectProto.hasOwnProperty;
    var maskSrcKey = function() {
      var uid = /[^.]+$/.exec(coreJsData && coreJsData.keys && coreJsData.keys.IE_PROTO || "");
      return uid ? "Symbol(src)_1." + uid : "";
    }();
    var nativeObjectToString = objectProto.toString;
    var objectCtorString = funcToString.call(Object);
    var reIsNative = RegExp(
      "^" + funcToString.call(hasOwnProperty).replace(reRegExpChar, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
    );
    var Buffer2 = moduleExports ? root.Buffer : void 0;
    var Symbol2 = root.Symbol;
    var Uint8Array2 = root.Uint8Array;
    var allocUnsafe = Buffer2 ? Buffer2.allocUnsafe : void 0;
    var getPrototype = overArg(Object.getPrototypeOf, Object);
    var objectCreate = Object.create;
    var propertyIsEnumerable = objectProto.propertyIsEnumerable;
    var splice = arrayProto.splice;
    var symToStringTag = Symbol2 ? Symbol2.toStringTag : void 0;
    var defineProperty = function() {
      try {
        var func = getNative(Object, "defineProperty");
        func({}, "", {});
        return func;
      } catch (e) {
      }
    }();
    var nativeIsBuffer = Buffer2 ? Buffer2.isBuffer : void 0;
    var nativeMax = Math.max;
    var nativeNow = Date.now;
    var Map2 = getNative(root, "Map");
    var nativeCreate = getNative(Object, "create");
    var baseCreate = /* @__PURE__ */ function() {
      function object() {
      }
      return function(proto) {
        if (!isObject2(proto)) {
          return {};
        }
        if (objectCreate) {
          return objectCreate(proto);
        }
        object.prototype = proto;
        var result = new object();
        object.prototype = void 0;
        return result;
      };
    }();
    function Hash(entries) {
      var index = -1, length = entries == null ? 0 : entries.length;
      this.clear();
      while (++index < length) {
        var entry = entries[index];
        this.set(entry[0], entry[1]);
      }
    }
    function hashClear() {
      this.__data__ = nativeCreate ? nativeCreate(null) : {};
      this.size = 0;
    }
    function hashDelete(key) {
      var result = this.has(key) && delete this.__data__[key];
      this.size -= result ? 1 : 0;
      return result;
    }
    function hashGet(key) {
      var data = this.__data__;
      if (nativeCreate) {
        var result = data[key];
        return result === HASH_UNDEFINED ? void 0 : result;
      }
      return hasOwnProperty.call(data, key) ? data[key] : void 0;
    }
    function hashHas(key) {
      var data = this.__data__;
      return nativeCreate ? data[key] !== void 0 : hasOwnProperty.call(data, key);
    }
    function hashSet(key, value) {
      var data = this.__data__;
      this.size += this.has(key) ? 0 : 1;
      data[key] = nativeCreate && value === void 0 ? HASH_UNDEFINED : value;
      return this;
    }
    Hash.prototype.clear = hashClear;
    Hash.prototype["delete"] = hashDelete;
    Hash.prototype.get = hashGet;
    Hash.prototype.has = hashHas;
    Hash.prototype.set = hashSet;
    function ListCache(entries) {
      var index = -1, length = entries == null ? 0 : entries.length;
      this.clear();
      while (++index < length) {
        var entry = entries[index];
        this.set(entry[0], entry[1]);
      }
    }
    function listCacheClear() {
      this.__data__ = [];
      this.size = 0;
    }
    function listCacheDelete(key) {
      var data = this.__data__, index = assocIndexOf(data, key);
      if (index < 0) {
        return false;
      }
      var lastIndex = data.length - 1;
      if (index == lastIndex) {
        data.pop();
      } else {
        splice.call(data, index, 1);
      }
      --this.size;
      return true;
    }
    function listCacheGet(key) {
      var data = this.__data__, index = assocIndexOf(data, key);
      return index < 0 ? void 0 : data[index][1];
    }
    function listCacheHas(key) {
      return assocIndexOf(this.__data__, key) > -1;
    }
    function listCacheSet(key, value) {
      var data = this.__data__, index = assocIndexOf(data, key);
      if (index < 0) {
        ++this.size;
        data.push([key, value]);
      } else {
        data[index][1] = value;
      }
      return this;
    }
    ListCache.prototype.clear = listCacheClear;
    ListCache.prototype["delete"] = listCacheDelete;
    ListCache.prototype.get = listCacheGet;
    ListCache.prototype.has = listCacheHas;
    ListCache.prototype.set = listCacheSet;
    function MapCache(entries) {
      var index = -1, length = entries == null ? 0 : entries.length;
      this.clear();
      while (++index < length) {
        var entry = entries[index];
        this.set(entry[0], entry[1]);
      }
    }
    function mapCacheClear() {
      this.size = 0;
      this.__data__ = {
        "hash": new Hash(),
        "map": new (Map2 || ListCache)(),
        "string": new Hash()
      };
    }
    function mapCacheDelete(key) {
      var result = getMapData(this, key)["delete"](key);
      this.size -= result ? 1 : 0;
      return result;
    }
    function mapCacheGet(key) {
      return getMapData(this, key).get(key);
    }
    function mapCacheHas(key) {
      return getMapData(this, key).has(key);
    }
    function mapCacheSet(key, value) {
      var data = getMapData(this, key), size = data.size;
      data.set(key, value);
      this.size += data.size == size ? 0 : 1;
      return this;
    }
    MapCache.prototype.clear = mapCacheClear;
    MapCache.prototype["delete"] = mapCacheDelete;
    MapCache.prototype.get = mapCacheGet;
    MapCache.prototype.has = mapCacheHas;
    MapCache.prototype.set = mapCacheSet;
    function Stack2(entries) {
      var data = this.__data__ = new ListCache(entries);
      this.size = data.size;
    }
    function stackClear() {
      this.__data__ = new ListCache();
      this.size = 0;
    }
    function stackDelete(key) {
      var data = this.__data__, result = data["delete"](key);
      this.size = data.size;
      return result;
    }
    function stackGet(key) {
      return this.__data__.get(key);
    }
    function stackHas(key) {
      return this.__data__.has(key);
    }
    function stackSet(key, value) {
      var data = this.__data__;
      if (data instanceof ListCache) {
        var pairs = data.__data__;
        if (!Map2 || pairs.length < LARGE_ARRAY_SIZE - 1) {
          pairs.push([key, value]);
          this.size = ++data.size;
          return this;
        }
        data = this.__data__ = new MapCache(pairs);
      }
      data.set(key, value);
      this.size = data.size;
      return this;
    }
    Stack2.prototype.clear = stackClear;
    Stack2.prototype["delete"] = stackDelete;
    Stack2.prototype.get = stackGet;
    Stack2.prototype.has = stackHas;
    Stack2.prototype.set = stackSet;
    function arrayLikeKeys(value, inherited) {
      var isArr = isArray(value), isArg = !isArr && isArguments(value), isBuff = !isArr && !isArg && isBuffer(value), isType = !isArr && !isArg && !isBuff && isTypedArray(value), skipIndexes = isArr || isArg || isBuff || isType, result = skipIndexes ? baseTimes(value.length, String) : [], length = result.length;
      for (var key in value) {
        if ((inherited || hasOwnProperty.call(value, key)) && !(skipIndexes && // Safari 9 has enumerable `arguments.length` in strict mode.
        (key == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
        isBuff && (key == "offset" || key == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
        isType && (key == "buffer" || key == "byteLength" || key == "byteOffset") || // Skip index properties.
        isIndex(key, length)))) {
          result.push(key);
        }
      }
      return result;
    }
    function assignMergeValue(object, key, value) {
      if (value !== void 0 && !eq(object[key], value) || value === void 0 && !(key in object)) {
        baseAssignValue(object, key, value);
      }
    }
    function assignValue(object, key, value) {
      var objValue = object[key];
      if (!(hasOwnProperty.call(object, key) && eq(objValue, value)) || value === void 0 && !(key in object)) {
        baseAssignValue(object, key, value);
      }
    }
    function assocIndexOf(array, key) {
      var length = array.length;
      while (length--) {
        if (eq(array[length][0], key)) {
          return length;
        }
      }
      return -1;
    }
    function baseAssignValue(object, key, value) {
      if (key == "__proto__" && defineProperty) {
        defineProperty(object, key, {
          "configurable": true,
          "enumerable": true,
          "value": value,
          "writable": true
        });
      } else {
        object[key] = value;
      }
    }
    var baseFor = createBaseFor();
    function baseGetTag(value) {
      if (value == null) {
        return value === void 0 ? undefinedTag : nullTag;
      }
      return symToStringTag && symToStringTag in Object(value) ? getRawTag(value) : objectToString(value);
    }
    function baseIsArguments(value) {
      return isObjectLike(value) && baseGetTag(value) == argsTag;
    }
    function baseIsNative(value) {
      if (!isObject2(value) || isMasked(value)) {
        return false;
      }
      var pattern = isFunction(value) ? reIsNative : reIsHostCtor;
      return pattern.test(toSource(value));
    }
    function baseIsTypedArray(value) {
      return isObjectLike(value) && isLength(value.length) && !!typedArrayTags[baseGetTag(value)];
    }
    function baseKeysIn(object) {
      if (!isObject2(object)) {
        return nativeKeysIn(object);
      }
      var isProto = isPrototype(object), result = [];
      for (var key in object) {
        if (!(key == "constructor" && (isProto || !hasOwnProperty.call(object, key)))) {
          result.push(key);
        }
      }
      return result;
    }
    function baseMerge(object, source, srcIndex, customizer, stack) {
      if (object === source) {
        return;
      }
      baseFor(source, function(srcValue, key) {
        stack || (stack = new Stack2());
        if (isObject2(srcValue)) {
          baseMergeDeep(object, source, key, srcIndex, baseMerge, customizer, stack);
        } else {
          var newValue = customizer ? customizer(safeGet(object, key), srcValue, key + "", object, source, stack) : void 0;
          if (newValue === void 0) {
            newValue = srcValue;
          }
          assignMergeValue(object, key, newValue);
        }
      }, keysIn);
    }
    function baseMergeDeep(object, source, key, srcIndex, mergeFunc, customizer, stack) {
      var objValue = safeGet(object, key), srcValue = safeGet(source, key), stacked = stack.get(srcValue);
      if (stacked) {
        assignMergeValue(object, key, stacked);
        return;
      }
      var newValue = customizer ? customizer(objValue, srcValue, key + "", object, source, stack) : void 0;
      var isCommon = newValue === void 0;
      if (isCommon) {
        var isArr = isArray(srcValue), isBuff = !isArr && isBuffer(srcValue), isTyped = !isArr && !isBuff && isTypedArray(srcValue);
        newValue = srcValue;
        if (isArr || isBuff || isTyped) {
          if (isArray(objValue)) {
            newValue = objValue;
          } else if (isArrayLikeObject(objValue)) {
            newValue = copyArray(objValue);
          } else if (isBuff) {
            isCommon = false;
            newValue = cloneBuffer(srcValue, true);
          } else if (isTyped) {
            isCommon = false;
            newValue = cloneTypedArray(srcValue, true);
          } else {
            newValue = [];
          }
        } else if (isPlainObject(srcValue) || isArguments(srcValue)) {
          newValue = objValue;
          if (isArguments(objValue)) {
            newValue = toPlainObject(objValue);
          } else if (!isObject2(objValue) || isFunction(objValue)) {
            newValue = initCloneObject(srcValue);
          }
        } else {
          isCommon = false;
        }
      }
      if (isCommon) {
        stack.set(srcValue, newValue);
        mergeFunc(newValue, srcValue, srcIndex, customizer, stack);
        stack["delete"](srcValue);
      }
      assignMergeValue(object, key, newValue);
    }
    function baseRest(func, start) {
      return setToString(overRest(func, start, identity), func + "");
    }
    var baseSetToString = !defineProperty ? identity : function(func, string) {
      return defineProperty(func, "toString", {
        "configurable": true,
        "enumerable": false,
        "value": constant(string),
        "writable": true
      });
    };
    function cloneBuffer(buffer, isDeep) {
      if (isDeep) {
        return buffer.slice();
      }
      var length = buffer.length, result = allocUnsafe ? allocUnsafe(length) : new buffer.constructor(length);
      buffer.copy(result);
      return result;
    }
    function cloneArrayBuffer(arrayBuffer) {
      var result = new arrayBuffer.constructor(arrayBuffer.byteLength);
      new Uint8Array2(result).set(new Uint8Array2(arrayBuffer));
      return result;
    }
    function cloneTypedArray(typedArray, isDeep) {
      var buffer = isDeep ? cloneArrayBuffer(typedArray.buffer) : typedArray.buffer;
      return new typedArray.constructor(buffer, typedArray.byteOffset, typedArray.length);
    }
    function copyArray(source, array) {
      var index = -1, length = source.length;
      array || (array = Array(length));
      while (++index < length) {
        array[index] = source[index];
      }
      return array;
    }
    function copyObject(source, props, object, customizer) {
      var isNew = !object;
      object || (object = {});
      var index = -1, length = props.length;
      while (++index < length) {
        var key = props[index];
        var newValue = customizer ? customizer(object[key], source[key], key, object, source) : void 0;
        if (newValue === void 0) {
          newValue = source[key];
        }
        if (isNew) {
          baseAssignValue(object, key, newValue);
        } else {
          assignValue(object, key, newValue);
        }
      }
      return object;
    }
    function createAssigner(assigner) {
      return baseRest(function(object, sources) {
        var index = -1, length = sources.length, customizer = length > 1 ? sources[length - 1] : void 0, guard = length > 2 ? sources[2] : void 0;
        customizer = assigner.length > 3 && typeof customizer == "function" ? (length--, customizer) : void 0;
        if (guard && isIterateeCall(sources[0], sources[1], guard)) {
          customizer = length < 3 ? void 0 : customizer;
          length = 1;
        }
        object = Object(object);
        while (++index < length) {
          var source = sources[index];
          if (source) {
            assigner(object, source, index, customizer);
          }
        }
        return object;
      });
    }
    function createBaseFor(fromRight) {
      return function(object, iteratee, keysFunc) {
        var index = -1, iterable = Object(object), props = keysFunc(object), length = props.length;
        while (length--) {
          var key = props[fromRight ? length : ++index];
          if (iteratee(iterable[key], key, iterable) === false) {
            break;
          }
        }
        return object;
      };
    }
    function getMapData(map, key) {
      var data = map.__data__;
      return isKeyable(key) ? data[typeof key == "string" ? "string" : "hash"] : data.map;
    }
    function getNative(object, key) {
      var value = getValue(object, key);
      return baseIsNative(value) ? value : void 0;
    }
    function getRawTag(value) {
      var isOwn = hasOwnProperty.call(value, symToStringTag), tag = value[symToStringTag];
      try {
        value[symToStringTag] = void 0;
        var unmasked = true;
      } catch (e) {
      }
      var result = nativeObjectToString.call(value);
      if (unmasked) {
        if (isOwn) {
          value[symToStringTag] = tag;
        } else {
          delete value[symToStringTag];
        }
      }
      return result;
    }
    function initCloneObject(object) {
      return typeof object.constructor == "function" && !isPrototype(object) ? baseCreate(getPrototype(object)) : {};
    }
    function isIndex(value, length) {
      var type = typeof value;
      length = length == null ? MAX_SAFE_INTEGER : length;
      return !!length && (type == "number" || type != "symbol" && reIsUint.test(value)) && (value > -1 && value % 1 == 0 && value < length);
    }
    function isIterateeCall(value, index, object) {
      if (!isObject2(object)) {
        return false;
      }
      var type = typeof index;
      if (type == "number" ? isArrayLike(object) && isIndex(index, object.length) : type == "string" && index in object) {
        return eq(object[index], value);
      }
      return false;
    }
    function isKeyable(value) {
      var type = typeof value;
      return type == "string" || type == "number" || type == "symbol" || type == "boolean" ? value !== "__proto__" : value === null;
    }
    function isMasked(func) {
      return !!maskSrcKey && maskSrcKey in func;
    }
    function isPrototype(value) {
      var Ctor = value && value.constructor, proto = typeof Ctor == "function" && Ctor.prototype || objectProto;
      return value === proto;
    }
    function nativeKeysIn(object) {
      var result = [];
      if (object != null) {
        for (var key in Object(object)) {
          result.push(key);
        }
      }
      return result;
    }
    function objectToString(value) {
      return nativeObjectToString.call(value);
    }
    function overRest(func, start, transform2) {
      start = nativeMax(start === void 0 ? func.length - 1 : start, 0);
      return function() {
        var args = arguments, index = -1, length = nativeMax(args.length - start, 0), array = Array(length);
        while (++index < length) {
          array[index] = args[start + index];
        }
        index = -1;
        var otherArgs = Array(start + 1);
        while (++index < start) {
          otherArgs[index] = args[index];
        }
        otherArgs[start] = transform2(array);
        return apply(func, this, otherArgs);
      };
    }
    function safeGet(object, key) {
      if (key === "constructor" && typeof object[key] === "function") {
        return;
      }
      if (key == "__proto__") {
        return;
      }
      return object[key];
    }
    var setToString = shortOut(baseSetToString);
    function shortOut(func) {
      var count = 0, lastCalled = 0;
      return function() {
        var stamp = nativeNow(), remaining = HOT_SPAN - (stamp - lastCalled);
        lastCalled = stamp;
        if (remaining > 0) {
          if (++count >= HOT_COUNT) {
            return arguments[0];
          }
        } else {
          count = 0;
        }
        return func.apply(void 0, arguments);
      };
    }
    function toSource(func) {
      if (func != null) {
        try {
          return funcToString.call(func);
        } catch (e) {
        }
        try {
          return func + "";
        } catch (e) {
        }
      }
      return "";
    }
    function eq(value, other) {
      return value === other || value !== value && other !== other;
    }
    var isArguments = baseIsArguments(/* @__PURE__ */ function() {
      return arguments;
    }()) ? baseIsArguments : function(value) {
      return isObjectLike(value) && hasOwnProperty.call(value, "callee") && !propertyIsEnumerable.call(value, "callee");
    };
    var isArray = Array.isArray;
    function isArrayLike(value) {
      return value != null && isLength(value.length) && !isFunction(value);
    }
    function isArrayLikeObject(value) {
      return isObjectLike(value) && isArrayLike(value);
    }
    var isBuffer = nativeIsBuffer || stubFalse;
    function isFunction(value) {
      if (!isObject2(value)) {
        return false;
      }
      var tag = baseGetTag(value);
      return tag == funcTag || tag == genTag || tag == asyncTag || tag == proxyTag;
    }
    function isLength(value) {
      return typeof value == "number" && value > -1 && value % 1 == 0 && value <= MAX_SAFE_INTEGER;
    }
    function isObject2(value) {
      var type = typeof value;
      return value != null && (type == "object" || type == "function");
    }
    function isObjectLike(value) {
      return value != null && typeof value == "object";
    }
    function isPlainObject(value) {
      if (!isObjectLike(value) || baseGetTag(value) != objectTag) {
        return false;
      }
      var proto = getPrototype(value);
      if (proto === null) {
        return true;
      }
      var Ctor = hasOwnProperty.call(proto, "constructor") && proto.constructor;
      return typeof Ctor == "function" && Ctor instanceof Ctor && funcToString.call(Ctor) == objectCtorString;
    }
    var isTypedArray = nodeIsTypedArray ? baseUnary(nodeIsTypedArray) : baseIsTypedArray;
    function toPlainObject(value) {
      return copyObject(value, keysIn(value));
    }
    function keysIn(object) {
      return isArrayLike(object) ? arrayLikeKeys(object, true) : baseKeysIn(object);
    }
    var mergeWith5 = createAssigner(function(object, source, srcIndex, customizer) {
      baseMerge(object, source, srcIndex, customizer);
    });
    function constant(value) {
      return function() {
        return value;
      };
    }
    function identity(value) {
      return value;
    }
    function stubFalse() {
      return false;
    }
    module.exports = mergeWith5;
  }
});

// src/index.ts
var src_exports = {};
__export(src_exports, {
  AbsoluteCenter: () => react.AbsoluteCenter,
  Accordion: () => react.Accordion,
  AccordionButton: () => react.AccordionButton,
  AccordionIcon: () => react.AccordionIcon,
  AccordionItem: () => react.AccordionItem,
  AccordionPanel: () => react.AccordionPanel,
  Alert: () => Alert,
  AlertDescription: () => react.AlertDescription,
  AlertDialog: () => react.AlertDialog,
  AlertDialogBody: () => react.AlertDialogBody,
  AlertDialogCloseButton: () => react.AlertDialogCloseButton,
  AlertDialogContent: () => react.AlertDialogContent,
  AlertDialogFooter: () => react.AlertDialogFooter,
  AlertDialogHeader: () => react.AlertDialogHeader,
  AlertDialogOverlay: () => react.AlertDialogOverlay,
  AlertIcon: () => react.AlertIcon,
  AlertTitle: () => react.AlertTitle,
  AspectRatio: () => react.AspectRatio,
  Avatar: () => react.Avatar,
  AvatarBadge: () => react.AvatarBadge,
  AvatarGroup: () => react.AvatarGroup,
  Badge: () => Badge,
  Box: () => react.Box,
  Breadcrumb: () => react.Breadcrumb,
  BreadcrumbItem: () => react.BreadcrumbItem,
  BreadcrumbLink: () => react.BreadcrumbLink,
  BreadcrumbSeparator: () => react.BreadcrumbSeparator,
  Button: () => Button,
  ButtonGroup: () => ButtonGroup,
  Card: () => Card,
  CardBody: () => react.CardBody,
  CardFooter: () => react.CardFooter,
  CardHeader: () => react.CardHeader,
  Center: () => react.Center,
  Checkbox: () => react.Checkbox,
  CheckboxGroup: () => react.CheckboxGroup,
  Circle: () => react.Circle,
  CircularProgress: () => react.CircularProgress,
  CircularProgressLabel: () => react.CircularProgressLabel,
  CloseButton: () => react.CloseButton,
  Code: () => react.Code,
  Collapse: () => react.Collapse,
  Container: () => react.Container,
  Datepicker: () => Datepicker,
  Divider: () => react.Divider,
  Drawer: () => react.Drawer,
  DrawerBody: () => react.DrawerBody,
  DrawerCloseButton: () => react.DrawerCloseButton,
  DrawerContent: () => react.DrawerContent,
  DrawerFooter: () => react.DrawerFooter,
  DrawerHeader: () => react.DrawerHeader,
  DrawerOverlay: () => react.DrawerOverlay,
  Editable: () => react.Editable,
  EditableInput: () => react.EditableInput,
  EditablePreview: () => react.EditablePreview,
  EditableTextarea: () => react.EditableTextarea,
  Fade: () => react.Fade,
  FileUpload: () => FileUpload,
  Flex: () => react.Flex,
  Footer: () => Footer,
  FooterInline: () => FooterInline,
  FormControl: () => FormControl,
  FormErrorMessage: () => FormErrorMessage,
  FormHelperText: () => FormHelperText,
  FormLabel: () => FormLabel,
  Grid: () => react.Grid,
  GridItem: () => react.GridItem,
  HStack: () => react.HStack,
  Header: () => Header,
  Heading: () => react.Heading,
  Hide: () => react.Hide,
  Highlight: () => react.Highlight,
  Icon: () => Icon,
  IconButton: () => IconButton,
  Image: () => react.Image,
  Input: () => react.Input,
  InputAddon: () => react.InputAddon,
  InputGroup: () => react.InputGroup,
  InputLeftAddon: () => react.InputLeftAddon,
  InputLeftElement: () => react.InputLeftElement,
  InputRightAddon: () => react.InputRightAddon,
  InputRightElement: () => react.InputRightElement,
  Kbd: () => react.Kbd,
  KvibProvider: () => KvibProvider,
  Link: () => Link,
  LinkBox: () => react.LinkBox,
  LinkOverlay: () => react.LinkOverlay,
  List: () => react.List,
  ListIcon: () => react.ListIcon,
  ListItem: () => react.ListItem,
  Logo: () => Logo,
  Menu: () => react.Menu,
  MenuButton: () => react.MenuButton,
  MenuDivider: () => react.MenuDivider,
  MenuGroup: () => react.MenuGroup,
  MenuItem: () => react.MenuItem,
  MenuItemOption: () => react.MenuItemOption,
  MenuList: () => react.MenuList,
  MenuOptionGroup: () => react.MenuOptionGroup,
  Modal: () => react.Modal,
  ModalBody: () => react.ModalBody,
  ModalCloseButton: () => react.ModalCloseButton,
  ModalContent: () => react.ModalContent,
  ModalFooter: () => react.ModalFooter,
  ModalHeader: () => react.ModalHeader,
  ModalOverlay: () => react.ModalOverlay,
  NumberInput: () => NumberInput,
  NumberInputField: () => NumberInputField,
  NumberInputStepper: () => NumberInputStepper,
  OrderedList: () => react.OrderedList,
  PinInput: () => PinInput,
  PinInputField: () => PinInputField,
  Popover: () => react.Popover,
  PopoverAnchor: () => react.PopoverAnchor,
  PopoverArrow: () => react.PopoverArrow,
  PopoverBody: () => react.PopoverBody,
  PopoverCloseButton: () => react.PopoverCloseButton,
  PopoverContent: () => react.PopoverContent,
  PopoverFooter: () => react.PopoverFooter,
  PopoverHeader: () => react.PopoverHeader,
  PopoverTrigger: () => react.PopoverTrigger,
  Portal: () => react.Portal,
  Progress: () => react.Progress,
  Radio: () => react.Radio,
  RadioGroup: () => react.RadioGroup,
  RangeSlider: () => react.RangeSlider,
  RangeSliderFilledTrack: () => react.RangeSliderFilledTrack,
  RangeSliderMark: () => react.RangeSliderMark,
  RangeSliderThumb: () => react.RangeSliderThumb,
  RangeSliderTrack: () => react.RangeSliderTrack,
  ScaleFade: () => react.ScaleFade,
  Search: () => Search,
  SearchAsync: () => SearchAsync,
  Select: () => react.Select,
  Show: () => react.Show,
  SimpleGrid: () => react.SimpleGrid,
  Skeleton: () => react.Skeleton,
  SkeletonCircle: () => react.SkeletonCircle,
  SkeletonText: () => react.SkeletonText,
  SkipNavContent: () => skipNav.SkipNavContent,
  SkipNavLink: () => skipNav.SkipNavLink,
  Slide: () => react.Slide,
  SlideFade: () => react.SlideFade,
  Slider: () => react.Slider,
  SliderFilledTrack: () => react.SliderFilledTrack,
  SliderMark: () => react.SliderMark,
  SliderThumb: () => react.SliderThumb,
  SliderTrack: () => react.SliderTrack,
  Spacer: () => react.Spacer,
  Spinner: () => react.Spinner,
  Square: () => react.Square,
  Stack: () => react.Stack,
  StackDivider: () => react.StackDivider,
  Stat: () => react.Stat,
  StatArrow: () => react.StatArrow,
  StatGroup: () => react.StatGroup,
  StatHelpText: () => react.StatHelpText,
  StatLabel: () => react.StatLabel,
  StatNumber: () => react.StatNumber,
  Step: () => react.Step,
  StepDescription: () => react.StepDescription,
  StepIcon: () => react.StepIcon,
  StepIndicator: () => react.StepIndicator,
  StepNumber: () => react.StepNumber,
  StepSeparator: () => react.StepSeparator,
  StepStatus: () => react.StepStatus,
  StepTitle: () => react.StepTitle,
  Stepper: () => react.Stepper,
  Switch: () => react.Switch,
  Tab: () => react.Tab,
  TabList: () => react.TabList,
  TabPanel: () => react.TabPanel,
  TabPanels: () => react.TabPanels,
  Table: () => react.Table,
  TableCaption: () => react.TableCaption,
  TableContainer: () => react.TableContainer,
  Tabs: () => Tabs,
  Tag: () => react.Tag,
  TagCloseButton: () => react.TagCloseButton,
  TagLabel: () => react.TagLabel,
  TagLeftIcon: () => react.TagLeftIcon,
  TagRightIcon: () => react.TagRightIcon,
  Tbody: () => react.Tbody,
  Td: () => react.Td,
  Text: () => react.Text,
  Textarea: () => react.Textarea,
  Tfoot: () => react.Tfoot,
  Th: () => react.Th,
  Thead: () => react.Thead,
  Timepicker: () => Timepicker,
  Tooltip: () => react.Tooltip,
  Tr: () => react.Tr,
  UnorderedList: () => react.UnorderedList,
  VStack: () => react.VStack,
  VisuallyHidden: () => react.VisuallyHidden,
  VisuallyHiddenInput: () => react.VisuallyHiddenInput,
  Wrap: () => react.Wrap,
  WrapItem: () => react.WrapItem,
  defaultKvibTheme: () => defaultKvibTheme,
  extendTheme: () => react.extendTheme,
  forwardRef: () => react.forwardRef,
  getCurrentTime: () => getCurrentTime,
  getTimestampFromTime: () => getTimestampFromTime,
  useBoolean: () => react.useBoolean,
  useBreakpointValue: () => react.useBreakpointValue,
  useClipboard: () => react.useClipboard,
  useConst: () => react.useConst,
  useControllableProp: () => react.useControllableProp,
  useControllableState: () => react.useControllableState,
  useDimensions: () => react.useDimensions,
  useDisclosure: () => react.useDisclosure,
  useMediaQuery: () => react.useMediaQuery,
  useMergeRefs: () => react.useMergeRefs,
  useOutsideClick: () => react.useOutsideClick,
  usePrefersReducedMotion: () => react.usePrefersReducedMotion,
  useSteps: () => react.useSteps,
  useTheme: () => react.useTheme,
  useToast: () => react.useToast,
  useToken: () => react.useToken,
  withDefaultColorScheme: () => react.withDefaultColorScheme
});
var Alert = react.forwardRef((_a, ref) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ jsxRuntime.jsx(react.Alert, __spreadProps(__spreadValues({}, props), { ref }));
});
var Badge = react.forwardRef((_a, ref) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ jsxRuntime.jsx(react.Badge, __spreadProps(__spreadValues({}, props), { ref }));
});

// src/icon/index.ts
var icon_exports = {};
__export(icon_exports, {
  Icon: () => Icon
});
var Icon = react.forwardRef(
  ({ icon, size, color: color2, weight, grade, isFilled = false, ariaIsHidden = false, className = "" }, ref) => {
    return /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        ref,
        className: `material-symbols-rounded ${className}`,
        "aria-hidden": ariaIsHidden,
        style: {
          width: size,
          fontSize: size,
          color: color2,
          fontVariationSettings: `'FILL' ${isFilled ? 1 : 0}, 'wght' ${weight ? weight : 300}, 'GRAD' ${grade ? grade : 0}`
        },
        children: icon
      }
    );
  }
);

// src/icon/index.ts
__reExport(icon_exports, material_symbols_star__namespace);
var Button = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { children, iconFill, isDisabled, isLoading, leftIcon, rightIcon, iconAriaIsHidden } = _b, props = __objRest(_b, ["children", "iconFill", "isDisabled", "isLoading", "leftIcon", "rightIcon", "iconAriaIsHidden"]);
    return /* @__PURE__ */ jsxRuntime.jsxs(react.Button, __spreadProps(__spreadValues({}, props), { ref, isDisabled: isDisabled || isLoading, "aria-busy": isLoading, children: [
      isLoading && /* @__PURE__ */ jsxRuntime.jsx(react.Center, { position: "absolute", right: "0", left: "0", children: /* @__PURE__ */ jsxRuntime.jsx(react.Spinner, { size: "sm" }) }),
      /* @__PURE__ */ jsxRuntime.jsxs(react.HStack, { spacing: 1, visibility: isLoading ? "hidden" : "visible", children: [
        leftIcon && /* @__PURE__ */ jsxRuntime.jsx(
          Icon,
          {
            icon: leftIcon,
            ariaIsHidden: iconAriaIsHidden,
            isFilled: iconFill,
            size: props.size === "xs" || props.size === "sm" ? 20 : 24
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(react.Center, { className: "text", as: "span", children }),
        rightIcon && /* @__PURE__ */ jsxRuntime.jsx(
          Icon,
          {
            icon: rightIcon,
            ariaIsHidden: iconAriaIsHidden,
            isFilled: iconFill,
            size: props.size === "xs" || props.size === "sm" ? 20 : 24
          }
        )
      ] })
    ] }));
  }
);
var IconSpinner = (props) => {
  if (props.isLoading) {
    return /* @__PURE__ */ jsxRuntime.jsx(react.Spinner, { size: "sm" });
  }
  return /* @__PURE__ */ jsxRuntime.jsx(
    Icon,
    {
      icon: props.icon,
      isFilled: props.iconFill,
      size: props.size === "xs" || props.size === "sm" ? 20 : 24,
      weight: props.size === "xs" || props.size === "sm" ? 300 : 400
    }
  );
};
var IconButton = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { isDisabled, isLoading, iconFill } = _b, props = __objRest(_b, ["isDisabled", "isLoading", "iconFill"]);
    return /* @__PURE__ */ jsxRuntime.jsx(
      react.IconButton,
      __spreadProps(__spreadValues({}, props), {
        ref,
        isDisabled: isDisabled || isLoading,
        "aria-busy": isLoading,
        icon: IconSpinner(__spreadValues({ isLoading, iconFill }, props))
      })
    );
  }
);
var ButtonGroup = react.forwardRef((_a, ref) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ jsxRuntime.jsx(react.ButtonGroup, __spreadProps(__spreadValues({}, props), { ref }));
});
var Card = react.forwardRef((_a, ref) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ jsxRuntime.jsx(react.Card, __spreadProps(__spreadValues({}, props), { ref }));
});
var FormControl = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { isRequired, isInvalid, isDisabled, isReadOnly, children } = _b, props = __objRest(_b, ["isRequired", "isInvalid", "isDisabled", "isReadOnly", "children"]);
    return /* @__PURE__ */ jsxRuntime.jsx(
      react.FormControl,
      __spreadProps(__spreadValues({}, props), {
        ref,
        isRequired,
        isInvalid,
        isDisabled,
        isReadOnly,
        children
      })
    );
  }
);
var FormLabel = react.forwardRef((_a, ref) => {
  var _b = _a, { label, children } = _b, props = __objRest(_b, ["label", "children"]);
  return /* @__PURE__ */ jsxRuntime.jsxs(react.FormLabel, __spreadProps(__spreadValues({}, props), { ref, fontWeight: "bold", children: [
    label,
    children
  ] }));
});
var FormHelperText = react.forwardRef((_a, ref) => {
  var _b = _a, { helperText, children } = _b, props = __objRest(_b, ["helperText", "children"]);
  return /* @__PURE__ */ jsxRuntime.jsxs(react.FormHelperText, __spreadProps(__spreadValues({}, props), { ref, fontSize: "16px", paddingBottom: "8px", children: [
    helperText,
    children
  ] }));
});
var FormErrorMessage = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { errorMessage, children } = _b, props = __objRest(_b, ["errorMessage", "children"]);
    return /* @__PURE__ */ jsxRuntime.jsxs(
      react.FormErrorMessage,
      __spreadProps(__spreadValues({}, props), {
        ref,
        background: "red.50",
        padding: "8px",
        borderStyle: "solid",
        borderColor: "red.100",
        borderWidth: "2px",
        borderRadius: "8px",
        gap: "5px",
        children: [
          /* @__PURE__ */ jsxRuntime.jsx("span", { className: "material-symbols-outlined", children: "Error" }),
          /* @__PURE__ */ jsxRuntime.jsxs(react.Text, { children: [
            errorMessage,
            children
          ] })
        ]
      })
    );
  }
);

// src/index.ts
__reExport(src_exports, icon_exports);
var Link = react.forwardRef((_a, ref) => {
  var _b = _a, { children } = _b, props = __objRest(_b, ["children"]);
  var _a2;
  const isExternal = props.isExternal !== void 0 ? props.isExternal : Boolean((_a2 = props.href) == null ? void 0 : _a2.match(/^https?:\/\//));
  return /* @__PURE__ */ jsxRuntime.jsxs(react.Link, __spreadProps(__spreadValues({}, props), { ref, isExternal, children: [
    children,
    isExternal && /* @__PURE__ */ jsxRuntime.jsx(
      "span",
      {
        className: "material-symbols-rounded",
        style: { fontSize: "18px", verticalAlign: "middle", marginLeft: "4px" },
        role: "link",
        "aria-hidden": true,
        children: "launch"
      }
    )
  ] }));
});
var NumberInput = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { isInvalid, isDisabled, children } = _b, props = __objRest(_b, ["isInvalid", "isDisabled", "children"]);
    return /* @__PURE__ */ jsxRuntime.jsx(react.NumberInput, __spreadProps(__spreadValues({}, props), { ref, isInvalid, isDisabled, children }));
  }
);
var NumberInputField = react.forwardRef((_a, ref) => {
  var _b = _a, { children } = _b, props = __objRest(_b, ["children"]);
  return /* @__PURE__ */ jsxRuntime.jsx(react.NumberInputField, __spreadProps(__spreadValues({}, props), { ref, children }));
});
var NumberInputStepper = react.forwardRef((_a, ref) => {
  var _b = _a, props = __objRest(_b, ["children"]);
  return /* @__PURE__ */ jsxRuntime.jsxs(react.NumberInputStepper, __spreadProps(__spreadValues({}, props), { ref, children: [
    /* @__PURE__ */ jsxRuntime.jsx(react.NumberIncrementStepper, {}),
    /* @__PURE__ */ jsxRuntime.jsx(react.NumberDecrementStepper, {})
  ] }));
});
var PinInput = (_a) => {
  var _b = _a, { size, children } = _b, props = __objRest(_b, ["size", "children"]);
  return /* @__PURE__ */ jsxRuntime.jsx(react.PinInput, __spreadProps(__spreadValues({}, props), { size, children }));
};
var PinInputField = react.forwardRef((_a, ref) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ jsxRuntime.jsx(react.PinInputField, __spreadProps(__spreadValues({}, props), { ref }));
});
var KvibProvider = (_a) => {
  var _b = _a, { theme = defaultKvibTheme, children } = _b, props = __objRest(_b, ["theme", "children"]);
  return /* @__PURE__ */ jsxRuntime.jsx(react.ChakraProvider, __spreadProps(__spreadValues({}, props), { theme, children }));
};
var Search = react.forwardRef(
  (_a, ref) => {
    var _b = _a, {
      id,
      colorScheme,
      size,
      variant,
      isDisabled,
      searchButton = "none",
      buttonWidth,
      buttonVariant = "tertiary",
      buttonText,
      role = "search"
    } = _b, props = __objRest(_b, [
      "id",
      "colorScheme",
      "size",
      "variant",
      "isDisabled",
      "searchButton",
      "buttonWidth",
      "buttonVariant",
      "buttonText",
      "role"
    ]);
    const elementRef = react$1.useRef(null);
    const dimensions = react.useDimensions(elementRef);
    const RenderButton = ({ position: position2 }) => {
      const borderRadiusProps = position2 === "left" ? { borderBottomRightRadius: 0, borderTopRightRadius: 0 } : { borderBottomLeftRadius: 0, borderTopLeftRadius: 0 };
      const buttonProps = __spreadProps(__spreadValues({}, borderRadiusProps), {
        colorScheme,
        variant: buttonVariant,
        isDisabled,
        width: buttonWidth,
        size,
        borderRadius: size === "sm" || size === "xs" ? "0.125rem" : void 0
      });
      return buttonText ? /* @__PURE__ */ jsxRuntime.jsx(Button, __spreadProps(__spreadValues({ ref: elementRef }, buttonProps), { rightIcon: "search", type: "submit", "aria-label": "search", children: buttonText })) : /* @__PURE__ */ jsxRuntime.jsx(IconButton, __spreadProps(__spreadValues({}, buttonProps), { "aria-label": "search", type: "submit", icon: "search" }));
    };
    const inputPadding = buttonWidth ? `calc(${buttonWidth} + 0.5rem)` : buttonText && dimensions ? `calc(${dimensions.borderBox.width}px + 0.5rem)` : "3rem";
    const RenderInputGroup = () => /* @__PURE__ */ jsxRuntime.jsxs(react.InputGroup, { size, width: props.width, children: [
      /* @__PURE__ */ jsxRuntime.jsx(
        react.Input,
        __spreadProps(__spreadValues({}, props), {
          id,
          ref,
          role,
          size,
          variant,
          isDisabled,
          paddingLeft: searchButton === "left" ? inputPadding : void 0,
          paddingRight: searchButton === "right" ? inputPadding : void 0
        })
      ),
      searchButton === "left" && /* @__PURE__ */ jsxRuntime.jsx(react.InputLeftElement, { width: "auto", children: /* @__PURE__ */ jsxRuntime.jsx(RenderButton, { position: "left" }) }),
      searchButton === "right" && /* @__PURE__ */ jsxRuntime.jsx(react.InputRightElement, { width: "auto", children: /* @__PURE__ */ jsxRuntime.jsx(RenderButton, { position: "right" }) })
    ] });
    return /* @__PURE__ */ jsxRuntime.jsx(RenderInputGroup, {});
  }
);
var Tabs = react.forwardRef((_a, ref) => {
  var _b = _a, { size = "md" } = _b, props = __objRest(_b, ["size"]);
  return /* @__PURE__ */ jsxRuntime.jsx(react.Tabs, __spreadProps(__spreadValues({ size }, props), { ref }));
});

// src/theme/components/index.ts
var components_exports = {};
__export(components_exports, {
  Alert: () => alertTheme,
  Breadcrumb: () => breadcrumbTheme,
  Button: () => buttonTheme,
  Checkbox: () => checkboxTheme,
  Datepicker: () => datepickerTheme,
  FileUpload: () => fileUploadTheme,
  IconButton: () => iconButtonTheme,
  Input: () => inputTheme,
  Link: () => linkTheme,
  Modal: () => modalTheme,
  NumberInput: () => numberInputTheme,
  Radio: () => radioTheme,
  Select: () => selectTheme,
  Slider: () => sliderTheme,
  Switch: () => switchTheme,
  Tabs: () => tabsTheme,
  Textarea: () => textareaTheme,
  Timepicker: () => timepickerTheme
});

// src/theme/tokens/index.ts
var tokens_exports = {};
__export(tokens_exports, {
  blur: () => blur,
  borders: () => borders,
  breakpoints: () => breakpoints,
  colors: () => colors,
  config: () => config,
  fontSizes: () => fontSizes,
  fontWeights: () => fontWeights,
  fonts: () => fonts,
  letterSpacings: () => letterSpacings,
  lineHeights: () => lineHeights,
  mapColors: () => mapColors,
  mapColorsRGBA: () => mapColorsRGBA,
  radii: () => radii,
  shadows: () => shadows,
  spacing: () => spacing,
  spacingOrder: () => spacingOrder,
  zIndices: () => zIndices
});

// src/theme/tokens/config.ts
var config = {
  cssVarPrefix: "kvib",
  initialColorMode: "light",
  useSystemColorMode: false
};

// src/theme/tokens/colors.ts
var colors = {
  green: {
    50: "#E9F4EC",
    100: "#D0ECD6",
    200: "#75C577",
    300: "#4DB450",
    400: "#1A833B",
    500: "#156630",
    600: "#05502A",
    700: "#044323",
    800: "#033019",
    900: "#021D0F"
  },
  blue: {
    50: "#E4F1F8",
    100: "#BAD7F8",
    200: "#66AEE6",
    300: "#3D92DE",
    400: "#2472C4",
    500: "#1A589F",
    600: "#213060",
    700: "#1A264C",
    800: "#141C39",
    900: "#0A0F1E"
  },
  gray: {
    50: "#F5F2F2",
    100: "#F0EDED",
    200: "#E3E0E0",
    300: "#D5D2D2",
    400: "#B5B2B2",
    500: "#707070",
    600: "#505050",
    700: "#404040",
    800: "#303030",
    900: "#202020"
  },
  red: {
    50: "#FCF1E8",
    100: "#F2D4C0",
    200: "#EBB095",
    300: "#E58869",
    400: "#D04A14",
    500: "#A32F00",
    600: "#4F1400",
    700: "#3D0000",
    800: "#2E0300",
    900: "#1A0000"
  },
  orange: {
    50: "#FFF9ED",
    100: "#FCEBCD",
    200: "#FFDD9D",
    300: "#FEC152",
    400: "#FEB026",
    500: "#E49713",
    600: "#B2750B",
    700: "#905F09",
    800: "#6A4607",
    900: "#432C04"
  },
  purple: {
    50: "#F7E6EB",
    100: "#E7C7E0",
    200: "#D9A1D3",
    300: "#D9A1D3",
    400: "#8C4799",
    500: "#70397B",
    600: "#512758",
    700: "#311735",
    800: "#211023",
    900: "#100812"
  },
  whiteAlpha: {
    50: "rgba(255, 255, 255, 0.04)",
    100: "rgba(255, 255, 255, 0.06)",
    200: "rgba(255, 255, 255, 0.08)",
    300: "rgba(255, 255, 255, 0.16)",
    400: "rgba(255, 255, 255, 0.24)",
    500: "rgba(255, 255, 255, 0.36)",
    600: "rgba(255, 255, 255, 0.48)",
    700: "rgba(255, 255, 255, 0.64)",
    800: "rgba(255, 255, 255, 0.80)",
    900: "rgba(255, 255, 255, 0.92)"
  },
  blackAlpha: {
    50: "rgba(0, 0, 0, 0.04)",
    100: "rgba(0, 0, 0, 0.06)",
    200: "rgba(0, 0, 0, 0.08)",
    300: "rgba(0, 0, 0, 0.16)",
    400: "rgba(0, 0, 0, 0.24)",
    500: "rgba(0, 0, 0, 0.36)",
    600: "rgba(0, 0, 0, 0.48)",
    700: "rgba(0, 0, 0, 0.64)",
    800: "rgba(0, 0, 0, 0.80)",
    900: "rgba(0, 0, 0, 0.92)"
  },
  black: "#000000",
  white: "#FFFFFF"
};

// src/theme/tokens/shadows.ts
var shadows = {
  xs: "0 0 0 1px rgba(0, 0, 0, 0.05)",
  sm: "0 1px 2px 0 rgba(0, 0, 0, 0.05)",
  base: "0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06)",
  md: "0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06)",
  lg: "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
  xl: "0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04)",
  "2xl": "0 25px 50px -12px rgba(0, 0, 0, 0.25)",
  outline: "0 0 0 3px rgba(66, 153, 225, 0.6)",
  inner: "inset 0 2px 4px 0 rgba(0,0,0,0.06)",
  none: "none",
  "dark-lg": "rgba(0, 0, 0, 0.1) 0px 0px 0px 1px, rgba(0, 0, 0, 0.2) 0px 5px 10px, rgba(0, 0, 0, 0.4) 0px 15px 40px"
};

// src/theme/tokens/radius.ts
var radii = {
  none: "0",
  sm: "0.125rem",
  base: "0.25rem",
  md: "0.375rem",
  lg: "0.5rem",
  xl: "0.75rem",
  "2xl": "1rem",
  "3xl": "1.5rem",
  full: "9999px"
};

// src/theme/tokens/breakpoints.ts
var breakpoints = {
  base: "0em",
  sm: "30em",
  md: "48em",
  lg: "62em",
  xl: "80em",
  "2xl": "96em"
};

// src/theme/tokens/borders.ts
var borders = {
  none: 0,
  "1px": "1px solid",
  "2px": "2px solid",
  "4px": "4px solid",
  "8px": "8px solid"
};

// src/theme/tokens/blur.ts
var blur = {
  none: 0,
  sm: "4px",
  base: "8px",
  md: "12px",
  lg: "16px",
  xl: "24px",
  "2xl": "40px",
  "3xl": "64px"
};

// src/theme/tokens/spacing.ts
var spacing = {
  px: "1px",
  0.5: "0.125rem",
  1: "0.25rem",
  1.5: "0.375rem",
  2: "0.5rem",
  2.5: "0.625rem",
  3: "0.75rem",
  3.5: "0.875rem",
  4: "1rem",
  5: "1.25rem",
  6: "1.5rem",
  7: "1.75rem",
  8: "2rem",
  9: "2.25rem",
  10: "2.5rem",
  12: "3rem",
  14: "3.5rem",
  16: "4rem",
  20: "5rem",
  24: "6rem",
  28: "7rem",
  32: "8rem",
  36: "9rem",
  40: "10rem",
  44: "11rem",
  48: "12rem",
  52: "13rem",
  56: "14rem",
  60: "15rem",
  64: "16rem",
  72: "18rem",
  80: "20rem",
  96: "24rem"
};
var spacingOrder = [
  // Define the order you want
  "px",
  0.5,
  1,
  1.5,
  2,
  2.5,
  3,
  3.5,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  12,
  14,
  16,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  72,
  80,
  96
];

// src/theme/tokens/typography.ts
var letterSpacings = {
  tighter: "-0.05em",
  tight: "-0.025em",
  normal: "0",
  wide: "0.025em",
  wider: "0.05em",
  widest: "0.1em"
};
var lineHeights = {
  normal: "normal",
  none: 1,
  shorter: 1.25,
  short: 1.375,
  base: 1.5,
  tall: 1.625,
  taller: "2",
  "3": ".75rem",
  "4": "1rem",
  "5": "1.25rem",
  "6": "1.5rem",
  "7": "1.75rem",
  "8": "2rem",
  "9": "2.25rem",
  "10": "2.5rem"
};
var fontWeights = {
  hairline: 100,
  thin: 200,
  light: 300,
  normal: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};
var fonts = {
  body: `'Mulish', sans-serif`,
  heading: `'Mulish', sans-serif`,
  mono: `SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace`
};
var fontSizes = {
  "3xs": "0.45rem",
  "2xs": "0.625rem",
  xs: "0.75rem",
  sm: "0.875rem",
  md: "1rem",
  lg: "1.125rem",
  xl: "1.25rem",
  "2xl": "1.5rem",
  "3xl": "1.875rem",
  "4xl": "2.25rem",
  "5xl": "3rem",
  "6xl": "3.75rem",
  "7xl": "4.5rem",
  "8xl": "6rem",
  "9xl": "8rem"
};

// src/theme/tokens/z-index.ts
var zIndices = {
  hide: -1,
  auto: "auto",
  base: 0,
  docked: 10,
  dropdown: 1e3,
  sticky: 1100,
  banner: 1200,
  overlay: 1300,
  modal: 1400,
  popover: 1500,
  skipLink: 1600,
  toast: 1700,
  tooltip: 1800
};

// src/theme/tokens/colorsMaps.ts
var mapColors = {
  blue: {
    1: "#476ED4",
    2: "#537EFF",
    3: "#3E8DF6",
    4: "#2BBCE0",
    5: "#00E3FF",
    6: "#00BEFF"
  },
  purple: {
    1: "#5952D2",
    2: "#AE7FE9",
    3: "#C0AFFB",
    4: "#D163E6",
    5: "#D163E6"
  },
  pink: {
    1: "#8A034F",
    2: "#B92659"
  },
  red: {
    1: "#E54848",
    2: "#F15D4E",
    3: "#FF9287",
    4: "#DF6E79"
  },
  brown: {
    1: "#90120A",
    2: "#9A5252",
    3: "#E1562C",
    4: "#E56D31",
    5: "#FFAE49"
  },
  yellow: {
    1: "#EBAB3B",
    2: "#EBAC23",
    3: "#D3C439",
    4: "#EFE645",
    5: "#FDDB56"
  },
  green: {
    1: "#005900",
    2: "#01A47D",
    3: "#00A76C",
    4: "#00CB85"
  },
  black: {
    1: "#000000",
    2: "#6C7980",
    3: "#C7D3D9"
  }
};
var mapColorsRGBA = {
  blue: {
    1: "rgba(71, 110, 212, 1)",
    2: "rgba(83, 126, 255, 1)",
    3: "rgba(62, 141, 246, 1)",
    4: "rgba(43, 188, 224, 1)",
    5: "rgba(0, 227, 255, 1)",
    6: "rgba(0, 190, 255, 1)"
  },
  purple: {
    1: "rgba(89, 82, 210, 1)",
    2: "rgba(174, 127, 233, 1)",
    3: "rgba(192, 175, 251, 1)",
    4: "rgba(209, 99, 230, 1)",
    5: "rgba(209, 99, 230, 1)"
  },
  pink: {
    1: "rgba(138, 3, 79, 1)",
    2: "rgba(185, 38, 89, 1)"
  },
  red: {
    1: "rgba(229, 72, 72, 1)",
    2: "rgba(241, 93, 78, 1)",
    3: "rgba(255, 146, 135, 1)",
    4: "rgba(223, 110, 121, 1)"
  },
  brown: {
    1: "rgba(144, 18, 10, 1)",
    2: "rgba(154, 82, 82, 1)",
    3: "rgba(225, 86, 44, 1)",
    4: "rgba(229, 109, 49, 1)",
    5: "rgba(255, 174, 73, 1)"
  },
  yellow: {
    1: "rgba(235, 171, 59, 1)",
    2: "rgba(235, 172, 35, 1)",
    3: "rgba(211, 196, 57, 1)",
    4: "rgba(239, 230, 69, 1)",
    5: "rgba(253, 219, 86, 1)"
  },
  green: {
    1: "rgba(0, 89, 0, 1)",
    2: "rgba(1, 164, 125, 1)",
    3: "rgba(0, 167, 108, 1)",
    4: "rgba(0, 203, 133, 1)"
  },
  black: {
    1: "rgba(0, 0, 0, 1)",
    2: "rgba(108, 121, 128, 1)",
    3: "rgba(199, 211, 217, 1)"
  }
};

// src/theme/components/link.ts
var linkTheme = react.defineStyleConfig({
  baseStyle: ({ colorScheme }) => ({
    textDecoration: "underline",
    _focusVisible: {
      outline: borders["2px"],
      borderRadius: "1px",
      outlineColor: colors.blue[600],
      boxShadow: "none"
    },
    color: `${colorScheme}.500`,
    _hover: {
      color: `${colorScheme}.400`,
      textDecoration: "none"
    },
    _active: {
      color: `${colorScheme}.600`,
      textDecoration: "underline"
    }
  })
});
var baseStyle = react.defineStyle({
  fontStyle: "normal",
  fontWeight: "600",
  gap: "8px",
  borderRadius: "6px",
  _disabled: {
    opacity: 0.5,
    cursor: "not-allowed"
  }
});
var variantSolid = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  if (c === "gray") {
    return {
      bg: colors.gray[100],
      color: colors.black,
      _hover: {
        bg: colors.gray[200],
        _disabled: {
          bg: colors.gray[100],
          color: colors.black
        }
      },
      _active: {
        bg: colors.gray[300],
        _disabled: {
          bg: colors.gray[100],
          color: colors.black
        }
      }
    };
  }
  return {
    bg: `${c}.500`,
    color: colors.white,
    _hover: {
      bg: `${c}.400`,
      _disabled: {
        bg: `${c}.500`,
        color: colors.white
      }
    },
    _active: {
      bg: `${c}.600`,
      _disabled: {
        bg: `${c}.500`,
        color: colors.white
      }
    }
  };
});
var variantOutline = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  if (c === "gray") {
    return {
      border: "1px",
      borderColor: colors.gray[200],
      color: colors.black,
      _hover: {
        bg: colors.gray[50],
        _disabled: {
          bg: colors.white,
          color: colors.black
        }
      },
      _active: {
        bg: colors.gray[100],
        _disabled: {
          bg: colors.white,
          color: colors.black
        }
      }
    };
  }
  return {
    border: "1px",
    borderColor: `${c}.500`,
    color: `${c}.500`,
    _hover: {
      borderColor: `${c}.400`,
      bg: `${c}.400`,
      color: colors.white,
      _disabled: {
        bg: colors.white,
        color: `${c}.500`,
        border: "1px"
      }
    },
    _active: {
      bg: `${c}.600`,
      borderColor: `${c}.600`,
      color: colors.white,
      _disabled: {
        bg: colors.white,
        color: `${c}.500`,
        border: "1px"
      }
    }
  };
});
var variantTertiary = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  if (c === "gray") {
    return {
      color: colors.gray[800],
      backgroundColor: "transparent",
      _hover: {
        textDecoration: "none",
        color: colors.gray[500],
        ".text": {
          textDecoration: "Underline"
        },
        _disabled: {
          ".text": {
            textDecoration: "none",
            color: colors.gray[800]
          }
        }
      },
      _active: {
        color: colors.gray[900],
        textDecoration: "none",
        ".text": {
          textDecoration: "Underline"
        },
        _disabled: {
          color: colors.gray[800]
        }
      },
      _loading: {
        color: colors.gray[800]
      }
    };
  }
  return {
    color: `${c}.500`,
    backgroundColor: "transparent",
    _hover: {
      textDecoration: "none",
      color: `${c}.400`,
      ".text": {
        textDecoration: "Underline"
      },
      _disabled: {
        ".text": {
          textDecoration: "none"
        },
        textDecoration: "none",
        color: `${c}.500`
      }
    },
    _active: {
      textDecoration: "none",
      color: `${c}.600`,
      ".text": {
        textDecoration: "underline"
      },
      _disabled: {
        ".text": {
          textDecoration: "none",
          color: `${c}.500`
        }
      }
    },
    _loading: {
      color: `${c}.500`
    }
  };
});
var variantGhost = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  if (c === "gray") {
    return {
      bg: "transparent",
      color: colors.gray[800],
      _hover: {
        bg: colors.gray[50],
        _disabled: {
          bg: "transparent",
          color: colors.gray[800]
        }
      },
      _active: {
        bg: colors.gray[100],
        color: colors.gray[900],
        _disabled: {
          bg: "transparent",
          color: colors.gray[800]
        }
      }
    };
  }
  return {
    bg: "transparent",
    color: `${c}.500`,
    _hover: {
      bg: `${c}.50`,
      color: `${c}.400`,
      _disabled: {
        bg: "transparent",
        color: `${c}.500`
      }
    },
    _active: {
      bg: `${c}.100`,
      color: `${c}.600`,
      _disabled: {
        bg: "transparent",
        color: `${c}.500`
      }
    }
  };
});
var variants = {
  primary: variantSolid,
  secondary: variantOutline,
  tertiary: variantTertiary,
  ghost: variantGhost
};
var buttonTheme = react.defineStyleConfig({
  baseStyle,
  variants,
  defaultProps: {
    variant: "primary",
    size: "md",
    colorScheme: "green"
  }
});

// ../../node_modules/@chakra-ui/anatomy/dist/chunk-OA3DH5LS.mjs
function anatomy(name, map = {}) {
  let called = false;
  function assert() {
    if (!called) {
      called = true;
      return;
    }
    throw new Error(
      "[anatomy] .part(...) should only be called once. Did you mean to use .extend(...) ?"
    );
  }
  function parts(...values) {
    assert();
    for (const part of values) {
      map[part] = toPart(part);
    }
    return anatomy(name, map);
  }
  function extend(...parts2) {
    for (const part of parts2) {
      if (part in map)
        continue;
      map[part] = toPart(part);
    }
    return anatomy(name, map);
  }
  function selectors() {
    const value = Object.fromEntries(
      Object.entries(map).map(([key, part]) => [key, part.selector])
    );
    return value;
  }
  function classnames() {
    const value = Object.fromEntries(
      Object.entries(map).map(([key, part]) => [key, part.className])
    );
    return value;
  }
  function toPart(part) {
    const el = ["container", "root"].includes(part != null ? part : "") ? [name] : [name, part];
    const attr = el.filter(Boolean).join("__");
    const className = `chakra-${attr}`;
    const partObj = {
      className,
      selector: `.${className}`,
      toString: () => part
    };
    return partObj;
  }
  const __type = {};
  return {
    parts,
    toPart,
    extend,
    selectors,
    classnames,
    get keys() {
      return Object.keys(map);
    },
    __type
  };
}

// ../../node_modules/@chakra-ui/anatomy/dist/chunk-7OOI6RFH.mjs
anatomy("accordion").parts("root", "container", "button", "panel").extend("icon");
var alertAnatomy = anatomy("alert").parts("title", "description", "container").extend("icon", "spinner");
anatomy("avatar").parts("label", "badge", "container").extend("excessLabel", "group");
var breadcrumbAnatomy = anatomy("breadcrumb").parts("link", "item", "container").extend("separator");
anatomy("button").parts();
anatomy("checkbox").parts("control", "icon", "container").extend("label");
anatomy("progress").parts("track", "filledTrack").extend("label");
anatomy("drawer").parts("overlay", "dialogContainer", "dialog").extend("header", "closeButton", "body", "footer");
anatomy("editable").parts(
  "preview",
  "input",
  "textarea"
);
anatomy("form").parts(
  "container",
  "requiredIndicator",
  "helperText"
);
anatomy("formError").parts("text", "icon");
anatomy("input").parts(
  "addon",
  "field",
  "element",
  "group"
);
anatomy("list").parts("container", "item", "icon");
anatomy("menu").parts("button", "list", "item").extend("groupTitle", "icon", "command", "divider");
var modalAnatomy = anatomy("modal").parts("overlay", "dialogContainer", "dialog").extend("header", "closeButton", "body", "footer");
anatomy("numberinput").parts(
  "root",
  "field",
  "stepperGroup",
  "stepper"
);
anatomy("pininput").parts("field");
anatomy("popover").parts("content", "header", "body", "footer").extend("popper", "arrow", "closeButton");
anatomy("progress").parts(
  "label",
  "filledTrack",
  "track"
);
anatomy("radio").parts(
  "container",
  "control",
  "label"
);
anatomy("select").parts("field", "icon");
anatomy("slider").parts(
  "container",
  "track",
  "thumb",
  "filledTrack",
  "mark"
);
anatomy("stat").parts(
  "container",
  "label",
  "helpText",
  "number",
  "icon"
);
anatomy("switch").parts(
  "container",
  "track",
  "thumb",
  "label"
);
anatomy("table").parts(
  "table",
  "thead",
  "tbody",
  "tr",
  "th",
  "td",
  "tfoot",
  "caption"
);
var tabsAnatomy = anatomy("tabs").parts(
  "root",
  "tab",
  "tablist",
  "tabpanel",
  "tabpanels",
  "indicator"
);
anatomy("tag").parts(
  "container",
  "label",
  "closeButton"
);
anatomy("card").parts(
  "container",
  "header",
  "body",
  "footer"
);
anatomy("stepper").parts(
  "stepper",
  "step",
  "title",
  "description",
  "indicator",
  "separator",
  "icon",
  "number"
);

// src/theme/components/tabs.ts
var helpers = react.createMultiStyleConfigHelpers(tabsAnatomy.keys);
var tabsTheme = helpers.defineMultiStyleConfig({
  baseStyle: ({ colorScheme }) => {
    return {
      root: {
        width: "fit-content"
      },
      tab: {
        color: colors.blue[900],
        _selected: {
          color: `${colorScheme}.500`
        },
        _hover: {
          color: `${colorScheme}.400`
        },
        _disabled: {
          color: colors.gray[400],
          _hover: {
            color: colors.gray[400]
          }
        }
      }
    };
  },
  defaultProps: {
    size: "md"
  }
});
var baseStyle2 = react.defineStyle({
  borderRadius: "6px",
  outline: 0,
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  flexShrink: 0,
  _focusVisible: {
    boxShadow: "outline"
  },
  _disabled: {
    opacity: 0.5,
    cursor: "not-allowed"
  }
});
var sizes = {
  xs: react.defineStyle({
    padding: "6px",
    gap: "6px",
    width: "24px",
    height: "24px"
  }),
  sm: react.defineStyle({
    padding: "9px",
    width: "32px",
    height: "32px"
  }),
  md: react.defineStyle({
    padding: "12px",
    width: "40px",
    height: "40px"
  }),
  lg: react.defineStyle({
    lineHeight: "28px",
    padding: "15px",
    width: "48px",
    height: "48px"
  })
};
var iconButtonTheme = react.defineStyleConfig({
  baseStyle: baseStyle2,
  sizes,
  defaultProps: {
    variant: "primary",
    size: "md"
  }
});
var inputTheme = react.defineStyleConfig({
  baseStyle: {
    // Any base styles should go directly here
  },
  variants: {
    outline: {
      field: {
        borderColor: "gray.400"
      }
    },
    flushed: {
      field: {
        borderColor: "gray.400"
      }
    }
  }
});
var selectTheme = react.defineStyleConfig({
  baseStyle: {
    // Any base styles should go directly here
  },
  variants: {
    outline: {
      field: {
        borderColor: "gray.400"
      }
    },
    flushed: {
      field: {
        borderColor: "gray.400"
      }
    }
  }
});
var baseStyle3 = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  return {
    control: {
      borderColor: `${c}.500`,
      // Border color when unchecked
      _checked: {
        _hover: {
          borderColor: `${c}.400`,
          backgroundColor: `${c}.400`
        }
      },
      _invalid: {
        _checked: {
          backgroundColor: colors.red[500],
          _hover: {
            borderColor: colors.red[400],
            backgroundColor: colors.red[400]
          }
        }
      }
    }
  };
});
var checkboxTheme = react.defineStyleConfig({
  baseStyle: baseStyle3
});
var numberInputTheme = react.defineStyleConfig({
  baseStyle: {
    // Any base styles should go directly here
  },
  variants: {
    outline: {
      field: {
        borderColor: "gray.400"
      }
    },
    flushed: {
      field: {
        borderColor: "gray.400"
      }
    }
  }
});
var baseStyle4 = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  return {
    control: {
      borderColor: `${c}.500`,
      // Border color when unchecked
      _checked: {
        _hover: {
          borderColor: `${c}.400`,
          backgroundColor: `${c}.400`
        }
      },
      _invalid: {
        _checked: {
          backgroundColor: colors.red[500],
          _hover: {
            borderColor: colors.red[400],
            backgroundColor: colors.red[400]
          }
        }
      }
    }
  };
});
var radioTheme = react.defineStyleConfig({
  baseStyle: baseStyle4
});
var baseStyle5 = react.defineStyle((props) => {
  return {
    track: {
      bg: "gray.400"
    },
    thumb: {
      borderWidth: "2px",
      borderColor: `${props.colorScheme}.500`
    }
  };
});
var sliderTheme = react.defineStyleConfig({
  baseStyle: baseStyle5
});
var switchTheme = react.defineStyleConfig({
  baseStyle: ({ colorScheme }) => ({
    track: {
      bg: "gray.400",
      // Border color when unchecked
      _checked: {
        bg: `${colorScheme}.500`
      }
    }
  })
});
var textareaTheme = react.defineStyleConfig({
  baseStyle: {
    // Any base styles should go directly here
  },
  variants: {
    outline: {
      borderColor: "gray.400"
    },
    flushed: {
      borderColor: "gray.400"
    }
  }
});

// ../../node_modules/@chakra-ui/shared-utils/dist/index.mjs
function isObject(value) {
  const type = typeof value;
  return value != null && (type === "object" || type === "function") && !Array.isArray(value);
}

// ../../node_modules/@chakra-ui/styled-system/dist/index.mjs
__toESM(require_lodash(), 1);
__toESM(require_lodash(), 1);
var import_lodash3 = __toESM(require_lodash(), 1);
__toESM(require_lodash(), 1);
var isImportant = (value) => /!(important)?$/.test(value);
var withoutImportant = (value) => typeof value === "string" ? value.replace(/!(important)?$/, "").trim() : value;
var tokenToCSSVar = (scale, value) => (theme) => {
  const valueStr = String(value);
  const important = isImportant(valueStr);
  const valueWithoutImportant = withoutImportant(valueStr);
  const key = scale ? `${scale}.${valueWithoutImportant}` : valueWithoutImportant;
  let transformed = isObject(theme.__cssMap) && key in theme.__cssMap ? theme.__cssMap[key].varRef : value;
  transformed = withoutImportant(transformed);
  return important ? `${transformed} !important` : transformed;
};
function createTransform(options) {
  const { scale, transform: transform2, compose } = options;
  const fn = (value, theme) => {
    var _a;
    const _value = tokenToCSSVar(scale, value)(theme);
    let result = (_a = transform2 == null ? void 0 : transform2(_value, theme)) != null ? _a : _value;
    if (compose) {
      result = compose(result, theme);
    }
    return result;
  };
  return fn;
}
var pipe = (...fns) => (v) => fns.reduce((a, b) => b(a), v);
function toConfig(scale, transform2) {
  return (property) => {
    const result = { property, scale };
    result.transform = createTransform({
      scale,
      transform: transform2
    });
    return result;
  };
}
var getRtl = ({ rtl, ltr }) => (theme) => theme.direction === "rtl" ? rtl : ltr;
function logical(options) {
  const { property, scale, transform: transform2 } = options;
  return {
    scale,
    property: getRtl(property),
    transform: scale ? createTransform({
      scale,
      compose: transform2
    }) : transform2
  };
}
var transformTemplate = [
  "rotate(var(--chakra-rotate, 0))",
  "scaleX(var(--chakra-scale-x, 1))",
  "scaleY(var(--chakra-scale-y, 1))",
  "skewX(var(--chakra-skew-x, 0))",
  "skewY(var(--chakra-skew-y, 0))"
];
function getTransformTemplate() {
  return [
    "translateX(var(--chakra-translate-x, 0))",
    "translateY(var(--chakra-translate-y, 0))",
    ...transformTemplate
  ].join(" ");
}
function getTransformGpuTemplate() {
  return [
    "translate3d(var(--chakra-translate-x, 0), var(--chakra-translate-y, 0), 0)",
    ...transformTemplate
  ].join(" ");
}
var filterTemplate = {
  "--chakra-blur": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-brightness": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-contrast": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-grayscale": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-hue-rotate": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-invert": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-saturate": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-sepia": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-drop-shadow": "var(--chakra-empty,/*!*/ /*!*/)",
  filter: [
    "var(--chakra-blur)",
    "var(--chakra-brightness)",
    "var(--chakra-contrast)",
    "var(--chakra-grayscale)",
    "var(--chakra-hue-rotate)",
    "var(--chakra-invert)",
    "var(--chakra-saturate)",
    "var(--chakra-sepia)",
    "var(--chakra-drop-shadow)"
  ].join(" ")
};
var backdropFilterTemplate = {
  backdropFilter: [
    "var(--chakra-backdrop-blur)",
    "var(--chakra-backdrop-brightness)",
    "var(--chakra-backdrop-contrast)",
    "var(--chakra-backdrop-grayscale)",
    "var(--chakra-backdrop-hue-rotate)",
    "var(--chakra-backdrop-invert)",
    "var(--chakra-backdrop-opacity)",
    "var(--chakra-backdrop-saturate)",
    "var(--chakra-backdrop-sepia)"
  ].join(" "),
  "--chakra-backdrop-blur": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-brightness": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-contrast": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-grayscale": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-hue-rotate": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-invert": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-opacity": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-saturate": "var(--chakra-empty,/*!*/ /*!*/)",
  "--chakra-backdrop-sepia": "var(--chakra-empty,/*!*/ /*!*/)"
};
function getRingTemplate(value) {
  return {
    "--chakra-ring-offset-shadow": `var(--chakra-ring-inset) 0 0 0 var(--chakra-ring-offset-width) var(--chakra-ring-offset-color)`,
    "--chakra-ring-shadow": `var(--chakra-ring-inset) 0 0 0 calc(var(--chakra-ring-width) + var(--chakra-ring-offset-width)) var(--chakra-ring-color)`,
    "--chakra-ring-width": value,
    boxShadow: [
      `var(--chakra-ring-offset-shadow)`,
      `var(--chakra-ring-shadow)`,
      `var(--chakra-shadow, 0 0 #0000)`
    ].join(", ")
  };
}
var flexDirectionTemplate = {
  "row-reverse": {
    space: "--chakra-space-x-reverse",
    divide: "--chakra-divide-x-reverse"
  },
  "column-reverse": {
    space: "--chakra-space-y-reverse",
    divide: "--chakra-divide-y-reverse"
  }
};
var directionMap = {
  "to-t": "to top",
  "to-tr": "to top right",
  "to-r": "to right",
  "to-br": "to bottom right",
  "to-b": "to bottom",
  "to-bl": "to bottom left",
  "to-l": "to left",
  "to-tl": "to top left"
};
var valueSet = new Set(Object.values(directionMap));
var globalSet = /* @__PURE__ */ new Set([
  "none",
  "-moz-initial",
  "inherit",
  "initial",
  "revert",
  "unset"
]);
var trimSpace = (str) => str.trim();
function parseGradient(value, theme) {
  if (value == null || globalSet.has(value))
    return value;
  const prevent = isCSSFunction(value) || globalSet.has(value);
  if (!prevent)
    return `url('${value}')`;
  const regex = /(^[a-z-A-Z]+)\((.*)\)/g;
  const results = regex.exec(value);
  const type = results == null ? void 0 : results[1];
  const values = results == null ? void 0 : results[2];
  if (!type || !values)
    return value;
  const _type = type.includes("-gradient") ? type : `${type}-gradient`;
  const [maybeDirection, ...stops] = values.split(",").map(trimSpace).filter(Boolean);
  if ((stops == null ? void 0 : stops.length) === 0)
    return value;
  const direction = maybeDirection in directionMap ? directionMap[maybeDirection] : maybeDirection;
  stops.unshift(direction);
  const _values = stops.map((stop) => {
    if (valueSet.has(stop))
      return stop;
    const firstStop = stop.indexOf(" ");
    const [_color, _stop] = firstStop !== -1 ? [stop.substr(0, firstStop), stop.substr(firstStop + 1)] : [stop];
    const _stopOrFunc = isCSSFunction(_stop) ? _stop : _stop && _stop.split(" ");
    const key = `colors.${_color}`;
    const color2 = key in theme.__cssMap ? theme.__cssMap[key].varRef : _color;
    return _stopOrFunc ? [
      color2,
      ...Array.isArray(_stopOrFunc) ? _stopOrFunc : [_stopOrFunc]
    ].join(" ") : color2;
  });
  return `${_type}(${_values.join(", ")})`;
}
var isCSSFunction = (value) => {
  return typeof value === "string" && value.includes("(") && value.includes(")");
};
var gradientTransform = (value, theme) => parseGradient(value, theme != null ? theme : {});
function isCssVar(value) {
  return /^var\(--.+\)$/.test(value);
}
var analyzeCSSValue = (value) => {
  const num = parseFloat(value.toString());
  const unit = value.toString().replace(String(num), "");
  return { unitless: !unit, value: num, unit };
};
var wrap = (str) => (value) => `${str}(${value})`;
var transformFunctions = {
  filter(value) {
    return value !== "auto" ? value : filterTemplate;
  },
  backdropFilter(value) {
    return value !== "auto" ? value : backdropFilterTemplate;
  },
  ring(value) {
    return getRingTemplate(transformFunctions.px(value));
  },
  bgClip(value) {
    return value === "text" ? { color: "transparent", backgroundClip: "text" } : { backgroundClip: value };
  },
  transform(value) {
    if (value === "auto")
      return getTransformTemplate();
    if (value === "auto-gpu")
      return getTransformGpuTemplate();
    return value;
  },
  vh(value) {
    return value === "$100vh" ? "var(--chakra-vh)" : value;
  },
  px(value) {
    if (value == null)
      return value;
    const { unitless } = analyzeCSSValue(value);
    return unitless || typeof value === "number" ? `${value}px` : value;
  },
  fraction(value) {
    return !(typeof value === "number") || value > 1 ? value : `${value * 100}%`;
  },
  float(value, theme) {
    const map = { left: "right", right: "left" };
    return theme.direction === "rtl" ? map[value] : value;
  },
  degree(value) {
    if (isCssVar(value) || value == null)
      return value;
    const unitless = typeof value === "string" && !value.endsWith("deg");
    return typeof value === "number" || unitless ? `${value}deg` : value;
  },
  gradient: gradientTransform,
  blur: wrap("blur"),
  opacity: wrap("opacity"),
  brightness: wrap("brightness"),
  contrast: wrap("contrast"),
  dropShadow: wrap("drop-shadow"),
  grayscale: wrap("grayscale"),
  hueRotate: (value) => wrap("hue-rotate")(transformFunctions.degree(value)),
  invert: wrap("invert"),
  saturate: wrap("saturate"),
  sepia: wrap("sepia"),
  bgImage(value) {
    if (value == null)
      return value;
    const prevent = isCSSFunction(value) || globalSet.has(value);
    return !prevent ? `url(${value})` : value;
  },
  outline(value) {
    const isNoneOrZero = String(value) === "0" || String(value) === "none";
    return value !== null && isNoneOrZero ? { outline: "2px solid transparent", outlineOffset: "2px" } : { outline: value };
  },
  flexDirection(value) {
    var _a;
    const { space: space2, divide: divide2 } = (_a = flexDirectionTemplate[value]) != null ? _a : {};
    const result = { flexDirection: value };
    if (space2)
      result[space2] = 1;
    if (divide2)
      result[divide2] = 1;
    return result;
  }
};
var t = {
  borderWidths: toConfig("borderWidths"),
  borderStyles: toConfig("borderStyles"),
  colors: toConfig("colors"),
  borders: toConfig("borders"),
  gradients: toConfig("gradients", transformFunctions.gradient),
  radii: toConfig("radii", transformFunctions.px),
  space: toConfig("space", pipe(transformFunctions.vh, transformFunctions.px)),
  spaceT: toConfig("space", pipe(transformFunctions.vh, transformFunctions.px)),
  degreeT(property) {
    return { property, transform: transformFunctions.degree };
  },
  prop(property, scale, transform2) {
    return __spreadValues({
      property,
      scale
    }, scale && {
      transform: createTransform({ scale, transform: transform2 })
    });
  },
  propT(property, transform2) {
    return { property, transform: transform2 };
  },
  sizes: toConfig("sizes", pipe(transformFunctions.vh, transformFunctions.px)),
  sizesT: toConfig("sizes", pipe(transformFunctions.vh, transformFunctions.fraction)),
  shadows: toConfig("shadows"),
  logical,
  blur: toConfig("blur", transformFunctions.blur)
};
var background = {
  background: t.colors("background"),
  backgroundColor: t.colors("backgroundColor"),
  backgroundImage: t.gradients("backgroundImage"),
  backgroundSize: true,
  backgroundPosition: true,
  backgroundRepeat: true,
  backgroundAttachment: true,
  backgroundClip: { transform: transformFunctions.bgClip },
  bgSize: t.prop("backgroundSize"),
  bgPosition: t.prop("backgroundPosition"),
  bg: t.colors("background"),
  bgColor: t.colors("backgroundColor"),
  bgPos: t.prop("backgroundPosition"),
  bgRepeat: t.prop("backgroundRepeat"),
  bgAttachment: t.prop("backgroundAttachment"),
  bgGradient: t.gradients("backgroundImage"),
  bgClip: { transform: transformFunctions.bgClip }
};
Object.assign(background, {
  bgImage: background.backgroundImage,
  bgImg: background.backgroundImage
});
var border = {
  border: t.borders("border"),
  borderWidth: t.borderWidths("borderWidth"),
  borderStyle: t.borderStyles("borderStyle"),
  borderColor: t.colors("borderColor"),
  borderRadius: t.radii("borderRadius"),
  borderTop: t.borders("borderTop"),
  borderBlockStart: t.borders("borderBlockStart"),
  borderTopLeftRadius: t.radii("borderTopLeftRadius"),
  borderStartStartRadius: t.logical({
    scale: "radii",
    property: {
      ltr: "borderTopLeftRadius",
      rtl: "borderTopRightRadius"
    }
  }),
  borderEndStartRadius: t.logical({
    scale: "radii",
    property: {
      ltr: "borderBottomLeftRadius",
      rtl: "borderBottomRightRadius"
    }
  }),
  borderTopRightRadius: t.radii("borderTopRightRadius"),
  borderStartEndRadius: t.logical({
    scale: "radii",
    property: {
      ltr: "borderTopRightRadius",
      rtl: "borderTopLeftRadius"
    }
  }),
  borderEndEndRadius: t.logical({
    scale: "radii",
    property: {
      ltr: "borderBottomRightRadius",
      rtl: "borderBottomLeftRadius"
    }
  }),
  borderRight: t.borders("borderRight"),
  borderInlineEnd: t.borders("borderInlineEnd"),
  borderBottom: t.borders("borderBottom"),
  borderBlockEnd: t.borders("borderBlockEnd"),
  borderBottomLeftRadius: t.radii("borderBottomLeftRadius"),
  borderBottomRightRadius: t.radii("borderBottomRightRadius"),
  borderLeft: t.borders("borderLeft"),
  borderInlineStart: {
    property: "borderInlineStart",
    scale: "borders"
  },
  borderInlineStartRadius: t.logical({
    scale: "radii",
    property: {
      ltr: ["borderTopLeftRadius", "borderBottomLeftRadius"],
      rtl: ["borderTopRightRadius", "borderBottomRightRadius"]
    }
  }),
  borderInlineEndRadius: t.logical({
    scale: "radii",
    property: {
      ltr: ["borderTopRightRadius", "borderBottomRightRadius"],
      rtl: ["borderTopLeftRadius", "borderBottomLeftRadius"]
    }
  }),
  borderX: t.borders(["borderLeft", "borderRight"]),
  borderInline: t.borders("borderInline"),
  borderY: t.borders(["borderTop", "borderBottom"]),
  borderBlock: t.borders("borderBlock"),
  borderTopWidth: t.borderWidths("borderTopWidth"),
  borderBlockStartWidth: t.borderWidths("borderBlockStartWidth"),
  borderTopColor: t.colors("borderTopColor"),
  borderBlockStartColor: t.colors("borderBlockStartColor"),
  borderTopStyle: t.borderStyles("borderTopStyle"),
  borderBlockStartStyle: t.borderStyles("borderBlockStartStyle"),
  borderBottomWidth: t.borderWidths("borderBottomWidth"),
  borderBlockEndWidth: t.borderWidths("borderBlockEndWidth"),
  borderBottomColor: t.colors("borderBottomColor"),
  borderBlockEndColor: t.colors("borderBlockEndColor"),
  borderBottomStyle: t.borderStyles("borderBottomStyle"),
  borderBlockEndStyle: t.borderStyles("borderBlockEndStyle"),
  borderLeftWidth: t.borderWidths("borderLeftWidth"),
  borderInlineStartWidth: t.borderWidths("borderInlineStartWidth"),
  borderLeftColor: t.colors("borderLeftColor"),
  borderInlineStartColor: t.colors("borderInlineStartColor"),
  borderLeftStyle: t.borderStyles("borderLeftStyle"),
  borderInlineStartStyle: t.borderStyles("borderInlineStartStyle"),
  borderRightWidth: t.borderWidths("borderRightWidth"),
  borderInlineEndWidth: t.borderWidths("borderInlineEndWidth"),
  borderRightColor: t.colors("borderRightColor"),
  borderInlineEndColor: t.colors("borderInlineEndColor"),
  borderRightStyle: t.borderStyles("borderRightStyle"),
  borderInlineEndStyle: t.borderStyles("borderInlineEndStyle"),
  borderTopRadius: t.radii(["borderTopLeftRadius", "borderTopRightRadius"]),
  borderBottomRadius: t.radii([
    "borderBottomLeftRadius",
    "borderBottomRightRadius"
  ]),
  borderLeftRadius: t.radii(["borderTopLeftRadius", "borderBottomLeftRadius"]),
  borderRightRadius: t.radii([
    "borderTopRightRadius",
    "borderBottomRightRadius"
  ])
};
Object.assign(border, {
  rounded: border.borderRadius,
  roundedTop: border.borderTopRadius,
  roundedTopLeft: border.borderTopLeftRadius,
  roundedTopRight: border.borderTopRightRadius,
  roundedTopStart: border.borderStartStartRadius,
  roundedTopEnd: border.borderStartEndRadius,
  roundedBottom: border.borderBottomRadius,
  roundedBottomLeft: border.borderBottomLeftRadius,
  roundedBottomRight: border.borderBottomRightRadius,
  roundedBottomStart: border.borderEndStartRadius,
  roundedBottomEnd: border.borderEndEndRadius,
  roundedLeft: border.borderLeftRadius,
  roundedRight: border.borderRightRadius,
  roundedStart: border.borderInlineStartRadius,
  roundedEnd: border.borderInlineEndRadius,
  borderStart: border.borderInlineStart,
  borderEnd: border.borderInlineEnd,
  borderTopStartRadius: border.borderStartStartRadius,
  borderTopEndRadius: border.borderStartEndRadius,
  borderBottomStartRadius: border.borderEndStartRadius,
  borderBottomEndRadius: border.borderEndEndRadius,
  borderStartRadius: border.borderInlineStartRadius,
  borderEndRadius: border.borderInlineEndRadius,
  borderStartWidth: border.borderInlineStartWidth,
  borderEndWidth: border.borderInlineEndWidth,
  borderStartColor: border.borderInlineStartColor,
  borderEndColor: border.borderInlineEndColor,
  borderStartStyle: border.borderInlineStartStyle,
  borderEndStyle: border.borderInlineEndStyle
});
var color = {
  color: t.colors("color"),
  textColor: t.colors("color"),
  fill: t.colors("fill"),
  stroke: t.colors("stroke")
};
var effect = {
  boxShadow: t.shadows("boxShadow"),
  mixBlendMode: true,
  blendMode: t.prop("mixBlendMode"),
  backgroundBlendMode: true,
  bgBlendMode: t.prop("backgroundBlendMode"),
  opacity: true
};
Object.assign(effect, {
  shadow: effect.boxShadow
});
var filter = {
  filter: { transform: transformFunctions.filter },
  blur: t.blur("--chakra-blur"),
  brightness: t.propT("--chakra-brightness", transformFunctions.brightness),
  contrast: t.propT("--chakra-contrast", transformFunctions.contrast),
  hueRotate: t.propT("--chakra-hue-rotate", transformFunctions.hueRotate),
  invert: t.propT("--chakra-invert", transformFunctions.invert),
  saturate: t.propT("--chakra-saturate", transformFunctions.saturate),
  dropShadow: t.propT("--chakra-drop-shadow", transformFunctions.dropShadow),
  backdropFilter: { transform: transformFunctions.backdropFilter },
  backdropBlur: t.blur("--chakra-backdrop-blur"),
  backdropBrightness: t.propT(
    "--chakra-backdrop-brightness",
    transformFunctions.brightness
  ),
  backdropContrast: t.propT("--chakra-backdrop-contrast", transformFunctions.contrast),
  backdropHueRotate: t.propT(
    "--chakra-backdrop-hue-rotate",
    transformFunctions.hueRotate
  ),
  backdropInvert: t.propT("--chakra-backdrop-invert", transformFunctions.invert),
  backdropSaturate: t.propT("--chakra-backdrop-saturate", transformFunctions.saturate)
};
var flexbox = {
  alignItems: true,
  alignContent: true,
  justifyItems: true,
  justifyContent: true,
  flexWrap: true,
  flexDirection: { transform: transformFunctions.flexDirection },
  flex: true,
  flexFlow: true,
  flexGrow: true,
  flexShrink: true,
  flexBasis: t.sizes("flexBasis"),
  justifySelf: true,
  alignSelf: true,
  order: true,
  placeItems: true,
  placeContent: true,
  placeSelf: true,
  gap: t.space("gap"),
  rowGap: t.space("rowGap"),
  columnGap: t.space("columnGap")
};
Object.assign(flexbox, {
  flexDir: flexbox.flexDirection
});
var grid = {
  gridGap: t.space("gridGap"),
  gridColumnGap: t.space("gridColumnGap"),
  gridRowGap: t.space("gridRowGap"),
  gridColumn: true,
  gridRow: true,
  gridAutoFlow: true,
  gridAutoColumns: true,
  gridColumnStart: true,
  gridColumnEnd: true,
  gridRowStart: true,
  gridRowEnd: true,
  gridAutoRows: true,
  gridTemplate: true,
  gridTemplateColumns: true,
  gridTemplateRows: true,
  gridTemplateAreas: true,
  gridArea: true
};
var interactivity = {
  appearance: true,
  cursor: true,
  resize: true,
  userSelect: true,
  pointerEvents: true,
  outline: { transform: transformFunctions.outline },
  outlineOffset: true,
  outlineColor: t.colors("outlineColor")
};
var layout = {
  width: t.sizesT("width"),
  inlineSize: t.sizesT("inlineSize"),
  height: t.sizes("height"),
  blockSize: t.sizes("blockSize"),
  boxSize: t.sizes(["width", "height"]),
  minWidth: t.sizes("minWidth"),
  minInlineSize: t.sizes("minInlineSize"),
  minHeight: t.sizes("minHeight"),
  minBlockSize: t.sizes("minBlockSize"),
  maxWidth: t.sizes("maxWidth"),
  maxInlineSize: t.sizes("maxInlineSize"),
  maxHeight: t.sizes("maxHeight"),
  maxBlockSize: t.sizes("maxBlockSize"),
  overflow: true,
  overflowX: true,
  overflowY: true,
  overscrollBehavior: true,
  overscrollBehaviorX: true,
  overscrollBehaviorY: true,
  display: true,
  aspectRatio: true,
  hideFrom: {
    scale: "breakpoints",
    transform: (value, theme) => {
      var _a, _b, _c;
      const breakpoint = (_c = (_b = (_a = theme.__breakpoints) == null ? void 0 : _a.get(value)) == null ? void 0 : _b.minW) != null ? _c : value;
      const mq = `@media screen and (min-width: ${breakpoint})`;
      return { [mq]: { display: "none" } };
    }
  },
  hideBelow: {
    scale: "breakpoints",
    transform: (value, theme) => {
      var _a, _b, _c;
      const breakpoint = (_c = (_b = (_a = theme.__breakpoints) == null ? void 0 : _a.get(value)) == null ? void 0 : _b._minW) != null ? _c : value;
      const mq = `@media screen and (max-width: ${breakpoint})`;
      return { [mq]: { display: "none" } };
    }
  },
  verticalAlign: true,
  boxSizing: true,
  boxDecorationBreak: true,
  float: t.propT("float", transformFunctions.float),
  objectFit: true,
  objectPosition: true,
  visibility: true,
  isolation: true
};
Object.assign(layout, {
  w: layout.width,
  h: layout.height,
  minW: layout.minWidth,
  maxW: layout.maxWidth,
  minH: layout.minHeight,
  maxH: layout.maxHeight,
  overscroll: layout.overscrollBehavior,
  overscrollX: layout.overscrollBehaviorX,
  overscrollY: layout.overscrollBehaviorY
});
var list = {
  listStyleType: true,
  listStylePosition: true,
  listStylePos: t.prop("listStylePosition"),
  listStyleImage: true,
  listStyleImg: t.prop("listStyleImage")
};
function get(obj, path, fallback, index) {
  const key = typeof path === "string" ? path.split(".") : [path];
  for (index = 0; index < key.length; index += 1) {
    if (!obj)
      break;
    obj = obj[key[index]];
  }
  return obj === void 0 ? fallback : obj;
}
var memoize = (fn) => {
  const cache = /* @__PURE__ */ new WeakMap();
  const memoizedFn = (obj, path, fallback, index) => {
    if (typeof obj === "undefined") {
      return fn(obj, path, fallback);
    }
    if (!cache.has(obj)) {
      cache.set(obj, /* @__PURE__ */ new Map());
    }
    const map = cache.get(obj);
    if (map.has(path)) {
      return map.get(path);
    }
    const value = fn(obj, path, fallback, index);
    map.set(path, value);
    return value;
  };
  return memoizedFn;
};
var memoizedGet = memoize(get);
var srOnly = {
  border: "0px",
  clip: "rect(0, 0, 0, 0)",
  width: "1px",
  height: "1px",
  margin: "-1px",
  padding: "0px",
  overflow: "hidden",
  whiteSpace: "nowrap",
  position: "absolute"
};
var srFocusable = {
  position: "static",
  width: "auto",
  height: "auto",
  clip: "auto",
  padding: "0",
  margin: "0",
  overflow: "visible",
  whiteSpace: "normal"
};
var getWithPriority = (theme, key, styles) => {
  const result = {};
  const obj = memoizedGet(theme, key, {});
  for (const prop in obj) {
    const isInStyles = prop in styles && styles[prop] != null;
    if (!isInStyles)
      result[prop] = obj[prop];
  }
  return result;
};
var others = {
  srOnly: {
    transform(value) {
      if (value === true)
        return srOnly;
      if (value === "focusable")
        return srFocusable;
      return {};
    }
  },
  layerStyle: {
    processResult: true,
    transform: (value, theme, styles) => getWithPriority(theme, `layerStyles.${value}`, styles)
  },
  textStyle: {
    processResult: true,
    transform: (value, theme, styles) => getWithPriority(theme, `textStyles.${value}`, styles)
  },
  apply: {
    processResult: true,
    transform: (value, theme, styles) => getWithPriority(theme, value, styles)
  }
};
var position = {
  position: true,
  pos: t.prop("position"),
  zIndex: t.prop("zIndex", "zIndices"),
  inset: t.spaceT("inset"),
  insetX: t.spaceT(["left", "right"]),
  insetInline: t.spaceT("insetInline"),
  insetY: t.spaceT(["top", "bottom"]),
  insetBlock: t.spaceT("insetBlock"),
  top: t.spaceT("top"),
  insetBlockStart: t.spaceT("insetBlockStart"),
  bottom: t.spaceT("bottom"),
  insetBlockEnd: t.spaceT("insetBlockEnd"),
  left: t.spaceT("left"),
  insetInlineStart: t.logical({
    scale: "space",
    property: { ltr: "left", rtl: "right" }
  }),
  right: t.spaceT("right"),
  insetInlineEnd: t.logical({
    scale: "space",
    property: { ltr: "right", rtl: "left" }
  })
};
Object.assign(position, {
  insetStart: position.insetInlineStart,
  insetEnd: position.insetInlineEnd
});
var ring = {
  ring: { transform: transformFunctions.ring },
  ringColor: t.colors("--chakra-ring-color"),
  ringOffset: t.prop("--chakra-ring-offset-width"),
  ringOffsetColor: t.colors("--chakra-ring-offset-color"),
  ringInset: t.prop("--chakra-ring-inset")
};
var space = {
  margin: t.spaceT("margin"),
  marginTop: t.spaceT("marginTop"),
  marginBlockStart: t.spaceT("marginBlockStart"),
  marginRight: t.spaceT("marginRight"),
  marginInlineEnd: t.spaceT("marginInlineEnd"),
  marginBottom: t.spaceT("marginBottom"),
  marginBlockEnd: t.spaceT("marginBlockEnd"),
  marginLeft: t.spaceT("marginLeft"),
  marginInlineStart: t.spaceT("marginInlineStart"),
  marginX: t.spaceT(["marginInlineStart", "marginInlineEnd"]),
  marginInline: t.spaceT("marginInline"),
  marginY: t.spaceT(["marginTop", "marginBottom"]),
  marginBlock: t.spaceT("marginBlock"),
  padding: t.space("padding"),
  paddingTop: t.space("paddingTop"),
  paddingBlockStart: t.space("paddingBlockStart"),
  paddingRight: t.space("paddingRight"),
  paddingBottom: t.space("paddingBottom"),
  paddingBlockEnd: t.space("paddingBlockEnd"),
  paddingLeft: t.space("paddingLeft"),
  paddingInlineStart: t.space("paddingInlineStart"),
  paddingInlineEnd: t.space("paddingInlineEnd"),
  paddingX: t.space(["paddingInlineStart", "paddingInlineEnd"]),
  paddingInline: t.space("paddingInline"),
  paddingY: t.space(["paddingTop", "paddingBottom"]),
  paddingBlock: t.space("paddingBlock")
};
Object.assign(space, {
  m: space.margin,
  mt: space.marginTop,
  mr: space.marginRight,
  me: space.marginInlineEnd,
  marginEnd: space.marginInlineEnd,
  mb: space.marginBottom,
  ml: space.marginLeft,
  ms: space.marginInlineStart,
  marginStart: space.marginInlineStart,
  mx: space.marginX,
  my: space.marginY,
  p: space.padding,
  pt: space.paddingTop,
  py: space.paddingY,
  px: space.paddingX,
  pb: space.paddingBottom,
  pl: space.paddingLeft,
  ps: space.paddingInlineStart,
  paddingStart: space.paddingInlineStart,
  pr: space.paddingRight,
  pe: space.paddingInlineEnd,
  paddingEnd: space.paddingInlineEnd
});
var textDecoration = {
  textDecorationColor: t.colors("textDecorationColor"),
  textDecoration: true,
  textDecor: { property: "textDecoration" },
  textDecorationLine: true,
  textDecorationStyle: true,
  textDecorationThickness: true,
  textUnderlineOffset: true,
  textShadow: t.shadows("textShadow")
};
var transform = {
  clipPath: true,
  transform: t.propT("transform", transformFunctions.transform),
  transformOrigin: true,
  translateX: t.spaceT("--chakra-translate-x"),
  translateY: t.spaceT("--chakra-translate-y"),
  skewX: t.degreeT("--chakra-skew-x"),
  skewY: t.degreeT("--chakra-skew-y"),
  scaleX: t.prop("--chakra-scale-x"),
  scaleY: t.prop("--chakra-scale-y"),
  scale: t.prop(["--chakra-scale-x", "--chakra-scale-y"]),
  rotate: t.degreeT("--chakra-rotate")
};
var transition = {
  transition: true,
  transitionDelay: true,
  animation: true,
  willChange: true,
  transitionDuration: t.prop("transitionDuration", "transition.duration"),
  transitionProperty: t.prop("transitionProperty", "transition.property"),
  transitionTimingFunction: t.prop(
    "transitionTimingFunction",
    "transition.easing"
  )
};
var typography = {
  fontFamily: t.prop("fontFamily", "fonts"),
  fontSize: t.prop("fontSize", "fontSizes", transformFunctions.px),
  fontWeight: t.prop("fontWeight", "fontWeights"),
  lineHeight: t.prop("lineHeight", "lineHeights"),
  letterSpacing: t.prop("letterSpacing", "letterSpacings"),
  textAlign: true,
  fontStyle: true,
  textIndent: true,
  wordBreak: true,
  overflowWrap: true,
  textOverflow: true,
  textTransform: true,
  whiteSpace: true,
  isTruncated: {
    transform(value) {
      if (value === true) {
        return {
          overflow: "hidden",
          textOverflow: "ellipsis",
          whiteSpace: "nowrap"
        };
      }
    }
  },
  noOfLines: {
    static: {
      overflow: "hidden",
      textOverflow: "ellipsis",
      display: "-webkit-box",
      WebkitBoxOrient: "vertical",
      //@ts-ignore
      WebkitLineClamp: "var(--chakra-line-clamp)"
    },
    property: "--chakra-line-clamp"
  }
};
var scroll = {
  scrollBehavior: true,
  scrollSnapAlign: true,
  scrollSnapStop: true,
  scrollSnapType: true,
  // scroll margin
  scrollMargin: t.spaceT("scrollMargin"),
  scrollMarginTop: t.spaceT("scrollMarginTop"),
  scrollMarginBottom: t.spaceT("scrollMarginBottom"),
  scrollMarginLeft: t.spaceT("scrollMarginLeft"),
  scrollMarginRight: t.spaceT("scrollMarginRight"),
  scrollMarginX: t.spaceT(["scrollMarginLeft", "scrollMarginRight"]),
  scrollMarginY: t.spaceT(["scrollMarginTop", "scrollMarginBottom"]),
  // scroll padding
  scrollPadding: t.spaceT("scrollPadding"),
  scrollPaddingTop: t.spaceT("scrollPaddingTop"),
  scrollPaddingBottom: t.spaceT("scrollPaddingBottom"),
  scrollPaddingLeft: t.spaceT("scrollPaddingLeft"),
  scrollPaddingRight: t.spaceT("scrollPaddingRight"),
  scrollPaddingX: t.spaceT(["scrollPaddingLeft", "scrollPaddingRight"]),
  scrollPaddingY: t.spaceT(["scrollPaddingTop", "scrollPaddingBottom"])
};
function resolveReference(operand) {
  if (isObject(operand) && operand.reference) {
    return operand.reference;
  }
  return String(operand);
}
var toExpression = (operator, ...operands) => operands.map(resolveReference).join(` ${operator} `).replace(/calc/g, "");
var add = (...operands) => `calc(${toExpression("+", ...operands)})`;
var subtract = (...operands) => `calc(${toExpression("-", ...operands)})`;
var multiply = (...operands) => `calc(${toExpression("*", ...operands)})`;
var divide = (...operands) => `calc(${toExpression("/", ...operands)})`;
var negate = (x) => {
  const value = resolveReference(x);
  if (value != null && !Number.isNaN(parseFloat(value))) {
    return String(value).startsWith("-") ? String(value).slice(1) : `-${value}`;
  }
  return multiply(value, -1);
};
var calc = Object.assign(
  (x) => ({
    add: (...operands) => calc(add(x, ...operands)),
    subtract: (...operands) => calc(subtract(x, ...operands)),
    multiply: (...operands) => calc(multiply(x, ...operands)),
    divide: (...operands) => calc(divide(x, ...operands)),
    negate: () => calc(negate(x)),
    toString: () => x.toString()
  }),
  {
    add,
    subtract,
    multiply,
    divide,
    negate
  }
);
var state = {
  hover: (str, post) => `${str}:hover ${post}, ${str}[data-hover] ${post}`,
  focus: (str, post) => `${str}:focus ${post}, ${str}[data-focus] ${post}`,
  focusVisible: (str, post) => `${str}:focus-visible ${post}`,
  focusWithin: (str, post) => `${str}:focus-within ${post}`,
  active: (str, post) => `${str}:active ${post}, ${str}[data-active] ${post}`,
  disabled: (str, post) => `${str}:disabled ${post}, ${str}[data-disabled] ${post}`,
  invalid: (str, post) => `${str}:invalid ${post}, ${str}[data-invalid] ${post}`,
  checked: (str, post) => `${str}:checked ${post}, ${str}[data-checked] ${post}`,
  indeterminate: (str, post) => `${str}:indeterminate ${post}, ${str}[aria-checked=mixed] ${post}, ${str}[data-indeterminate] ${post}`,
  readOnly: (str, post) => `${str}:read-only ${post}, ${str}[readonly] ${post}, ${str}[data-read-only] ${post}`,
  expanded: (str, post) => `${str}:read-only ${post}, ${str}[aria-expanded=true] ${post}, ${str}[data-expanded] ${post}`,
  placeholderShown: (str, post) => `${str}:placeholder-shown ${post}`
};
var toGroup = (fn) => merge((v) => fn(v, "&"), "[role=group]", "[data-group]", ".group");
var toPeer = (fn) => merge((v) => fn(v, "~ &"), "[data-peer]", ".peer");
var merge = (fn, ...selectors) => selectors.map(fn).join(", ");
var pseudoSelectors = {
  /**
   * Styles for CSS selector `&:hover`
   */
  _hover: "&:hover, &[data-hover]",
  /**
   * Styles for CSS Selector `&:active`
   */
  _active: "&:active, &[data-active]",
  /**
   * Styles for CSS selector `&:focus`
   *
   */
  _focus: "&:focus, &[data-focus]",
  /**
   * Styles for the highlighted state.
   */
  _highlighted: "&[data-highlighted]",
  /**
   * Styles to apply when a child of this element has received focus
   * - CSS Selector `&:focus-within`
   */
  _focusWithin: "&:focus-within",
  /**
   * Styles to apply when this element has received focus via tabbing
   * - CSS Selector `&:focus-visible`
   */
  _focusVisible: "&:focus-visible, &[data-focus-visible]",
  /**
   * Styles to apply when this element is disabled. The passed styles are applied to these CSS selectors:
   * - `&[aria-disabled=true]`
   * - `&:disabled`
   * - `&[data-disabled]`
   * - `&[disabled]`
   */
  _disabled: "&:disabled, &[disabled], &[aria-disabled=true], &[data-disabled]",
  /**
   * Styles for CSS Selector `&:readonly`
   */
  _readOnly: "&[aria-readonly=true], &[readonly], &[data-readonly]",
  /**
   * Styles for CSS selector `&::before`
   *
   * NOTE:When using this, ensure the `content` is wrapped in a backtick.
   * @example
   * ```jsx
   * <Box _before={{content:`""` }}/>
   * ```
   */
  _before: "&::before",
  /**
   * Styles for CSS selector `&::after`
   *
   * NOTE:When using this, ensure the `content` is wrapped in a backtick.
   * @example
   * ```jsx
   * <Box _after={{content:`""` }}/>
   * ```
   */
  _after: "&::after",
  /**
   * Styles for CSS selector `&:empty`
   */
  _empty: "&:empty",
  /**
   * Styles to apply when the ARIA attribute `aria-expanded` is `true`
   * - CSS selector `&[aria-expanded=true]`
   */
  _expanded: "&[aria-expanded=true], &[data-expanded]",
  /**
   * Styles to apply when the ARIA attribute `aria-checked` is `true`
   * - CSS selector `&[aria-checked=true]`
   */
  _checked: "&[aria-checked=true], &[data-checked]",
  /**
   * Styles to apply when the ARIA attribute `aria-grabbed` is `true`
   * - CSS selector `&[aria-grabbed=true]`
   */
  _grabbed: "&[aria-grabbed=true], &[data-grabbed]",
  /**
   * Styles for CSS Selector `&[aria-pressed=true]`
   * Typically used to style the current "pressed" state of toggle buttons
   */
  _pressed: "&[aria-pressed=true], &[data-pressed]",
  /**
   * Styles to apply when the ARIA attribute `aria-invalid` is `true`
   * - CSS selector `&[aria-invalid=true]`
   */
  _invalid: "&[aria-invalid=true], &[data-invalid]",
  /**
   * Styles for the valid state
   * - CSS selector `&[data-valid], &[data-state=valid]`
   */
  _valid: "&[data-valid], &[data-state=valid]",
  /**
   * Styles for CSS Selector `&[aria-busy=true]` or `&[data-loading=true]`.
   * Useful for styling loading states
   */
  _loading: "&[data-loading], &[aria-busy=true]",
  /**
   * Styles to apply when the ARIA attribute `aria-selected` is `true`
   *
   * - CSS selector `&[aria-selected=true]`
   */
  _selected: "&[aria-selected=true], &[data-selected]",
  /**
   * Styles for CSS Selector `[hidden=true]`
   */
  _hidden: "&[hidden], &[data-hidden]",
  /**
   * Styles for CSS Selector `&:-webkit-autofill`
   */
  _autofill: "&:-webkit-autofill",
  /**
   * Styles for CSS Selector `&:nth-child(even)`
   */
  _even: "&:nth-of-type(even)",
  /**
   * Styles for CSS Selector `&:nth-child(odd)`
   */
  _odd: "&:nth-of-type(odd)",
  /**
   * Styles for CSS Selector `&:first-of-type`
   */
  _first: "&:first-of-type",
  /**
   * Styles for CSS selector `&::first-letter`
   *
   * NOTE: This selector is only applied for block-level elements and not preceded by an image or table.
   * @example
   * ```jsx
   * <Text _firstLetter={{ textDecoration: 'underline' }}>Once upon a time</Text>
   * ```
   */
  _firstLetter: "&::first-letter",
  /**
   * Styles for CSS Selector `&:last-of-type`
   */
  _last: "&:last-of-type",
  /**
   * Styles for CSS Selector `&:not(:first-of-type)`
   */
  _notFirst: "&:not(:first-of-type)",
  /**
   * Styles for CSS Selector `&:not(:last-of-type)`
   */
  _notLast: "&:not(:last-of-type)",
  /**
   * Styles for CSS Selector `&:visited`
   */
  _visited: "&:visited",
  /**
   * Used to style the active link in a navigation
   * Styles for CSS Selector `&[aria-current=page]`
   */
  _activeLink: "&[aria-current=page]",
  /**
   * Used to style the current step within a process
   * Styles for CSS Selector `&[aria-current=step]`
   */
  _activeStep: "&[aria-current=step]",
  /**
   * Styles to apply when the ARIA attribute `aria-checked` is `mixed`
   * - CSS selector `&[aria-checked=mixed]`
   */
  _indeterminate: "&:indeterminate, &[aria-checked=mixed], &[data-indeterminate]",
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` is hovered
   */
  _groupHover: toGroup(state.hover),
  /**
   * Styles to apply when a sibling element with `.peer` or `data-peer` is hovered
   */
  _peerHover: toPeer(state.hover),
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` is focused
   */
  _groupFocus: toGroup(state.focus),
  /**
   * Styles to apply when a sibling element with `.peer` or `data-peer` is focused
   */
  _peerFocus: toPeer(state.focus),
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` has visible focus
   */
  _groupFocusVisible: toGroup(state.focusVisible),
  /**
   * Styles to apply when a sibling element with `.peer`or `data-peer` has visible focus
   */
  _peerFocusVisible: toPeer(state.focusVisible),
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` is active
   */
  _groupActive: toGroup(state.active),
  /**
   * Styles to apply when a sibling element with `.peer` or `data-peer` is active
   */
  _peerActive: toPeer(state.active),
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` is disabled
   */
  _groupDisabled: toGroup(state.disabled),
  /**
   *  Styles to apply when a sibling element with `.peer` or `data-peer` is disabled
   */
  _peerDisabled: toPeer(state.disabled),
  /**
   *  Styles to apply when a parent element with `.group`, `data-group` or `role=group` is invalid
   */
  _groupInvalid: toGroup(state.invalid),
  /**
   *  Styles to apply when a sibling element with `.peer` or `data-peer` is invalid
   */
  _peerInvalid: toPeer(state.invalid),
  /**
   * Styles to apply when a parent element with `.group`, `data-group` or `role=group` is checked
   */
  _groupChecked: toGroup(state.checked),
  /**
   * Styles to apply when a sibling element with `.peer` or `data-peer` is checked
   */
  _peerChecked: toPeer(state.checked),
  /**
   *  Styles to apply when a parent element with `.group`, `data-group` or `role=group` has focus within
   */
  _groupFocusWithin: toGroup(state.focusWithin),
  /**
   *  Styles to apply when a sibling element with `.peer` or `data-peer` has focus within
   */
  _peerFocusWithin: toPeer(state.focusWithin),
  /**
   * Styles to apply when a sibling element with `.peer` or `data-peer` has placeholder shown
   */
  _peerPlaceholderShown: toPeer(state.placeholderShown),
  /**
   * Styles for CSS Selector `&::placeholder`.
   */
  _placeholder: "&::placeholder",
  /**
   * Styles for CSS Selector `&:placeholder-shown`.
   */
  _placeholderShown: "&:placeholder-shown",
  /**
   * Styles for CSS Selector `&:fullscreen`.
   */
  _fullScreen: "&:fullscreen",
  /**
   * Styles for CSS Selector `&::selection`
   */
  _selection: "&::selection",
  /**
   * Styles for CSS Selector `[dir=rtl] &`
   * It is applied when a parent element or this element has `dir="rtl"`
   */
  _rtl: "[dir=rtl] &, &[dir=rtl]",
  /**
   * Styles for CSS Selector `[dir=ltr] &`
   * It is applied when a parent element or this element has `dir="ltr"`
   */
  _ltr: "[dir=ltr] &, &[dir=ltr]",
  /**
   * Styles for CSS Selector `@media (prefers-color-scheme: dark)`
   * It is used when the user has requested the system use a light or dark color theme.
   */
  _mediaDark: "@media (prefers-color-scheme: dark)",
  /**
   * Styles for CSS Selector `@media (prefers-reduced-motion: reduce)`
   * It is used when the user has requested the system to reduce the amount of animations.
   */
  _mediaReduceMotion: "@media (prefers-reduced-motion: reduce)",
  /**
   * Styles for when `data-theme` is applied to any parent of
   * this component or element.
   */
  _dark: ".chakra-ui-dark &:not([data-theme]),[data-theme=dark] &:not([data-theme]),&[data-theme=dark]",
  /**
   * Styles for when `data-theme` is applied to any parent of
   * this component or element.
   */
  _light: ".chakra-ui-light &:not([data-theme]),[data-theme=light] &:not([data-theme]),&[data-theme=light]",
  /**
   * Styles for the CSS Selector `&[data-orientation=horizontal]`
   */
  _horizontal: "&[data-orientation=horizontal]",
  /**
   * Styles for the CSS Selector `&[data-orientation=vertical]`
   */
  _vertical: "&[data-orientation=vertical]"
};
var pseudoPropNames = Object.keys(
  pseudoSelectors
);
var systemProps = (0, import_lodash3.default)(
  {},
  background,
  border,
  color,
  flexbox,
  layout,
  filter,
  ring,
  interactivity,
  grid,
  others,
  position,
  effect,
  space,
  scroll,
  typography,
  textDecoration,
  transform,
  list,
  transition
);
Object.assign({}, space, layout, flexbox, grid, position);
[...Object.keys(systemProps), ...pseudoPropNames];
__spreadValues(__spreadValues({}, systemProps), pseudoSelectors);
function createMultiStyleConfigHelpers2(parts) {
  return {
    definePartsStyle(config2) {
      return config2;
    },
    defineMultiStyleConfig(config2) {
      return __spreadValues({ parts }, config2);
    }
  };
}

// src/theme/components/breadcrumb.ts
var helpers2 = createMultiStyleConfigHelpers2(breadcrumbAnatomy.keys);
var breadcrumbTheme = helpers2.defineMultiStyleConfig({
  baseStyle: ({ colorScheme }) => ({
    link: {
      "&:not([aria-current=page])": {
        color: `${colorScheme}.500`,
        textDecoration: "underline",
        _focusVisible: {
          outline: borders["2px"],
          borderRadius: "1px",
          outlineColor: colors.blue[600],
          boxShadow: "none"
        },
        _active: {
          color: `${colorScheme}.600`,
          textDecoration: "underline"
        },
        _hover: {
          color: `${colorScheme}.400`,
          textDecoration: "none"
        }
      }
    }
  })
});
var baseStyle6 = react.defineStyle((props) => {
  const { colorScheme: c } = props;
  return {
    bg: "transparent",
    color: "black",
    _hover: {
      backgroundColor: `${c}.50`,
      color: `${c}.400`,
      _disabled: {
        bg: "transparent",
        color: `${c}.500`
      }
    },
    _active: {
      bg: `${c}.100`,
      color: `${c}.600`,
      _disabled: {
        bg: "transparent",
        color: `${c}.500`
      }
    },
    _focus: {
      backgroundColor: `${c}.100`,
      color: `${c}.500`
    }
  };
});
var timepickerTheme = react.defineStyleConfig({
  baseStyle: baseStyle6,
  defaultProps: {
    variant: "outline"
  }
});
var sizes2 = {
  xs: react.defineStyle({
    padding: "1px"
  }),
  sm: react.defineStyle({
    padding: "2px"
  }),
  md: react.defineStyle({
    padding: "5px"
  }),
  lg: react.defineStyle({
    padding: "5px"
  })
};
var fileUploadTheme = react.defineStyleConfig({
  sizes: sizes2,
  defaultProps: {
    size: "md"
  }
});
var { defineMultiStyleConfig } = react.createMultiStyleConfigHelpers(alertAnatomy.keys);
var variantSubtle = react.defineStyle((props) => {
  const { status: s } = props;
  if (s === "error") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.red[100],
        color: colors.black
      },
      icon: {
        color: colors.red[500]
      }
    };
  }
  if (s === "info") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.blue[100],
        color: colors.black
      },
      icon: {
        color: colors.blue[500]
      }
    };
  }
  return {};
});
var variantSolid2 = react.defineStyle((props) => {
  const { status: s } = props;
  if (s === "error") {
    return {
      title: {
        color: colors.white
      },
      container: {
        bg: colors.red[500],
        color: colors.white
      },
      icon: {
        color: colors.white
      }
    };
  }
  if (s === "info") {
    return {
      title: {
        color: colors.white
      },
      container: {
        bg: colors.blue[500],
        color: colors.white
      },
      icon: {
        color: colors.white
      }
    };
  }
  return {};
});
var variantLeftAccent = react.defineStyle((props) => {
  const { status: s } = props;
  if (s === "error") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.red[100],
        color: colors.black,
        borderColor: colors.red[500]
      },
      icon: {
        color: colors.red[500]
      }
    };
  }
  if (s === "info") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.blue[100],
        color: colors.black,
        borderColor: colors.blue[500]
      },
      icon: {
        color: colors.blue[500]
      }
    };
  }
  return {};
});
var variantTopAccent = react.defineStyle((props) => {
  const { status: s } = props;
  if (s === "error") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.red[100],
        color: colors.black,
        borderColor: colors.red[500]
      },
      icon: {
        color: colors.red[500]
      }
    };
  }
  if (s === "info") {
    return {
      title: {
        color: colors.black
      },
      container: {
        bg: colors.blue[100],
        color: colors.black,
        borderColor: colors.blue[500]
      },
      icon: {
        color: colors.blue[500]
      }
    };
  }
  return {};
});
var variants2 = {
  subtle: variantSubtle,
  solid: variantSolid2,
  "left-accent": variantLeftAccent,
  "top-accent": variantTopAccent
};
var alertTheme = defineMultiStyleConfig({ variants: variants2 });
var datepickerTheme = react.defineStyleConfig({
  defaultProps: {
    colorScheme: "green"
  }
});

// src/theme/components/modal.ts
var { definePartsStyle, defineMultiStyleConfig: defineMultiStyleConfig2 } = createMultiStyleConfigHelpers2(modalAnatomy.keys);
var baseStyle7 = definePartsStyle({
  overlay: {
    bg: "blackAlpha.700"
  }
});
var modalTheme = defineMultiStyleConfig2({
  baseStyle: baseStyle7
});

// src/theme/index.ts
var defaultKvibTheme = react.extendTheme(
  react.withDefaultColorScheme({ colorScheme: "green" }),
  react.withDefaultColorScheme({ colorScheme: "gray", components: ["Badge", "Code", "Table", "Tag"] }),
  __spreadProps(__spreadValues({}, tokens_exports), {
    components: __spreadValues({}, components_exports)
  })
);
var Datepicker = react.forwardRef((_a, ref) => {
  var _b = _a, { onChange, useNative = true } = _b, props = __objRest(_b, ["onChange", "useNative"]);
  const KVInputProps = extractKVProps(props);
  const commonProps = getCommonInputProps(props);
  const defaultValue = props.defaultSelected ? formatDate(props.defaultSelected) : void 0;
  const isClient = typeof window === "object";
  const isMobile = isClient ? window.innerWidth < 480 : false;
  if (isMobile && useNative)
    return /* @__PURE__ */ jsxRuntime.jsx(
      react.Input,
      __spreadProps(__spreadValues(__spreadValues({
        ref,
        type: "date",
        defaultValue
      }, KVInputProps), commonProps), {
        onChange
      })
    );
  return /* @__PURE__ */ jsxRuntime.jsx(CustomDatepicker, __spreadProps(__spreadValues(__spreadValues({}, props), commonProps), { onChange }));
});
var CustomDatepicker = react.forwardRef(
  (_a, ref) => {
    var _b = _a, {
      onChange,
      defaultSelected,
      defaultMonth,
      fromDate,
      toDate,
      showDropdownMonthYear = true,
      disableNavigation,
      showOutsideDays,
      showWeekNumber,
      disabledDays,
      colorScheme,
      isDisabled: isDisabledExternally = false,
      isInvalid: isInvalidExternally = false,
      isRequired: isRequiredExternally = false
    } = _b, KVInputProps = __objRest(_b, [
      "onChange",
      "defaultSelected",
      "defaultMonth",
      "fromDate",
      "toDate",
      "showDropdownMonthYear",
      "disableNavigation",
      "showOutsideDays",
      "showWeekNumber",
      "disabledDays",
      "colorScheme",
      "isDisabled",
      "isInvalid",
      "isRequired"
    ]);
    const theme = react.useTheme();
    const uniqueClassName = generateUniqueClassName("kvib-datepicker");
    const style = css(
      uniqueClassName,
      theme.colors[colorScheme != null ? colorScheme : theme.components.Datepicker.defaultProps.colorScheme]
    );
    const formControlContext = react.useFormControlContext();
    const isDisabledFromForm = (formControlContext == null ? void 0 : formControlContext.isDisabled) || false;
    const isInvalidFromForm = (formControlContext == null ? void 0 : formControlContext.isInvalid) || false;
    const isRequiredFromForm = (formControlContext == null ? void 0 : formControlContext.isRequired) || false;
    const isDisabled = isDisabledExternally || isDisabledFromForm;
    const isInvalid = isInvalidExternally || isInvalidFromForm;
    const isRequired = isRequiredExternally || isRequiredFromForm;
    const [isPickerVisible, setPickerVisible] = react.useBoolean(false);
    const { inputProps, dayPickerProps } = reactDayPicker.useInput({
      defaultSelected,
      format: "dd.MM.yyyy",
      locale: nb.nb,
      fromDate,
      toDate,
      required: isRequired
    });
    react$1.useEffect(() => {
      if (dayPickerProps.selected) {
        effectiveOnChange({
          target: {
            value: formatDate(dayPickerProps.selected)
          },
          bubbles: true,
          type: "change",
          persist: () => {
          },
          nativeEvent: {}
        });
        setPickerVisible.off();
      }
    }, [dayPickerProps.selected, setPickerVisible]);
    const effectiveOnChange = (event) => {
      const dateStr = event.target.value;
      const parsedDate = dateFns.parse(dateStr, "yyyy-MM-dd", /* @__PURE__ */ new Date());
      if (dateFns.isValid(parsedDate)) {
        onChange == null ? void 0 : onChange(event);
      }
    };
    return /* @__PURE__ */ jsxRuntime.jsxs(
      react.Popover,
      {
        placement: "bottom-start",
        isOpen: isPickerVisible,
        onOpen: setPickerVisible.on,
        onClose: setPickerVisible.off,
        children: [
          /* @__PURE__ */ jsxRuntime.jsxs(react.InputGroup, { children: [
            /* @__PURE__ */ jsxRuntime.jsx(react.PopoverAnchor, { children: /* @__PURE__ */ jsxRuntime.jsx(
              react.Input,
              __spreadValues(__spreadValues({
                ref,
                className: "custom-datepicker",
                isDisabled,
                isInvalid,
                isRequired
              }, KVInputProps), inputProps)
            ) }),
            /* @__PURE__ */ jsxRuntime.jsx(react.InputRightElement, { opacity: isDisabled ? 0.5 : 1, pointerEvents: isDisabled ? "none" : "auto", height: "100%", children: /* @__PURE__ */ jsxRuntime.jsx(react.PopoverTrigger, { children: /* @__PURE__ */ jsxRuntime.jsx(
              IconButton,
              {
                icon: "calendar_today",
                colorScheme,
                size: KVInputProps.size,
                "aria-label": "open datepicker",
                onClick: setPickerVisible.toggle,
                variant: "tertiary"
              }
            ) }) })
          ] }),
          /* @__PURE__ */ jsxRuntime.jsxs(react.PopoverContent, { width: "auto", padding: "1rem", children: [
            /* @__PURE__ */ jsxRuntime.jsx("style", { children: style }),
            /* @__PURE__ */ jsxRuntime.jsx(
              reactDayPicker.DayPicker,
              __spreadValues({
                captionLayout: showDropdownMonthYear ? "dropdown-buttons" : void 0,
                disableNavigation,
                defaultMonth,
                showOutsideDays,
                showWeekNumber,
                disabled: disabledDays,
                classNames: { root: uniqueClassName }
              }, dayPickerProps)
            )
          ] })
        ]
      }
    );
  }
);
function extractKVProps(props) {
  const _a = props, chakraProps = __objRest(_a, [
    "defaultSelected",
    "defaultMonth",
    "fromDate",
    "toDate",
    "showDropdownMonthYear",
    "disableNavigation",
    "showOutsideDays",
    "showWeekNumber",
    "disabledDays"
  ]);
  return chakraProps;
}
function formatDate(date, format) {
  let dateObject;
  if (typeof date === "number") {
    dateObject = new Date(date);
  } else if (date instanceof Date) {
    dateObject = date;
  } else if (typeof date === "string") {
    dateObject = new Date(date);
    if (isNaN(dateObject.getTime())) {
      throw new Error("Invalid date format.");
    }
  } else {
    throw new Error("Invalid date format.");
  }
  const y = dateObject.getFullYear().toString().padStart(4, "0");
  const m = (dateObject.getMonth() + 1).toString().padStart(2, "0");
  const d = dateObject.getDate().toString().padStart(2, "0");
  return format === "no" ? `${d}.${m}.${y}` : `${y}-${m}-${d}`;
}
var getCommonInputProps = (props) => {
  const min = props.fromDate ? formatDate(props.fromDate) : void 0;
  const max = props.toDate ? formatDate(props.toDate) : void 0;
  return {
    min,
    max
  };
};
var css = (className, colorPalette) => {
  return `
 .${className} {
  --rdp-cell-size: 40px; /* Size of the day cells. */
  --rdp-caption-font-size: 18px; /* Font size for the caption labels. */
  --rdp-accent-color: ${colorPalette[500]}; /* Accent color for the background of selected days. */
  --rdp-background-color: ${colorPalette[50]}; /* Background color for the hovered/focused elements. */
  --rdp-outline: 2px solid var(--rdp-accent-color); /* Outline border for focused elements */
  --rdp-selected-color: #fff; /* Color of selected day text */
}
`;
};
var generateUniqueClassName = (baseName) => {
  const uniquePart = Math.random().toString(36).substring(2, 7);
  return `${baseName}-${uniquePart}`;
};
var FileUpload = react.forwardRef(
  (_a, ref) => {
    var _b = _a, { size, variant, type = "file", isRequired, isInvalid, isDisabled } = _b, props = __objRest(_b, ["size", "variant", "type", "isRequired", "isInvalid", "isDisabled"]);
    const styles = react.useStyleConfig("FileUpload", props);
    return /* @__PURE__ */ jsxRuntime.jsx(
      react.Input,
      __spreadProps(__spreadValues({}, props), {
        __css: __spreadValues({}, styles),
        ref,
        size,
        type,
        variant,
        isRequired,
        isInvalid,
        isDisabled
      })
    );
  }
);
var LogoHorizontal = ({ label, width, height }) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      "aria-label": label,
      width,
      height,
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 184 64",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M40.938 2.1h-34.6c-1.7 0-3.1 1.4-3.1 3.1v21.2c2.7-2.1 6.1-4.9 12.2-3.4 5.3 1.3 6.4 5.4 14.8 10 5.1 2.8 9.9.6 13.8-1.6V5.2c0-1.7-1.4-3.1-3.1-3.1z",
            fill: "#1A833B"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M32.938 42.4c-1.7-.2-3.1-.8-4.4-1.7-2.9-1.8-5.7-4.7-11.7-7.8-4.7-2.3-9.4-.5-13.6 1.9v23.7c0 1.7 1.4 3.1 3.1 3.1h34.5c1.7 0 3.1-1.4 3.1-3.1V39.1c-3.6 2.1-7.1 3.9-11 3.3z",
            fill: "#1A589F"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M30.238 32.9c-8.3-4.6-9.5-8.7-14.8-10-6.1-1.5-9.5 1.4-12.2 3.4V34.7c4.2-2.4 8.9-4.3 13.6-1.9 6 3.1 8.8 6 11.7 7.8 1.3.8 2.7 1.4 4.4 1.7 3.9.6 7.4-1.1 11.1-3.2V31.2c-3.9 2.4-8.7 4.6-13.8 1.7z",
            fill: "#fff"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M55.04 50.9V31.3h2.4v19.6h-2.4zm5.1-10.4 7.8 10.4h-3l-7.5-10.4 7.5-9.2h3l-7.8 9.2zM80.838 51.4c-1-.3-1.7-1.1-1.9-2.1-1.4 1.4-2.5 1.9-4.1 1.9-3.5 0-4.6-2.1-4.6-4.1 0-3.2 2.8-5.1 7.5-5.1h1v-1c0-1.3-.1-1.8-.5-2.2-.5-.5-1-.8-1.9-.8-1.6 0-3.6.9-4.5 1.7l-1.1-1.5c1.9-1.3 3.9-1.9 6-1.9 2 0 3.3.8 3.9 2.2.3.8.3 1.9.3 3.1l-.1 4.7c-.1 2.3.1 2.9 1.1 3.5l-1.1 1.6zm-3.2-7.8c-3.6 0-4.9 1.2-4.9 3.4 0 1.7.8 2.6 2.5 2.6 1.5 0 2.8-.9 3.5-2.2l.1-3.7c-.4-.1-.9-.1-1.2-.1zM92.14 38.8c-.1-.1-.4-.1-.6-.1-.9 0-1.7.4-2.4 1-.7.7-.8 1.1-.8 2.3v8.9h-2.2v-11c0-1.9-.4-2.8-.5-3l2.2-.6c.1.2.5 1.3.5 2.3 1-1.5 2.5-2.4 4-2.4.3 0 .7.1.8.1l-1 2.5zM102.239 38.3h-3v9.2c0 1.6.4 2.1 1.8 2.1.6 0 1-.1 1.4-.3l.3 1.4c-.7.4-1.5.6-2.5.6-.7 0-1.3-.1-1.9-.4-1-.5-1.4-1.4-1.4-2.8v-9.8h-1.9v-1.7h1.9c0-1.2.2-3.1.2-3.6l2.3-.5c-.1.5-.3 2.4-.3 4.1h3.6l-.5 1.7zM112.64 51h-2.2l-5.1-14.2 2.3-.5 3.3 9.8c.3 1 .6 2.2.7 2.4h.1c.1-.2.3-1.3.7-2.4l3.3-9.4h2.4l-5.5 14.3zM122.641 44.1v.3c0 1.3.2 2.4.7 3.1.8 1.2 2.1 1.7 3.5 1.7s2.5-.4 3.5-1.3l.9 1.4c-1.3 1.1-3 1.8-4.8 1.8-3.9 0-6.3-2.9-6.3-7.6 0-2.4.5-3.9 1.7-5.4 1.1-1.3 2.5-2 4.3-2 1.5 0 2.9.5 3.9 1.5 1.2 1.3 1.6 2.6 1.6 6v.3h-9v.2zm5.8-4.8c-.5-.8-1.5-1.3-2.6-1.3-2 0-3.2 1.5-3.3 4.4h6.5c.1-1.5-.1-2.4-.6-3.1zM141.738 38.8c-.1-.1-.4-.1-.6-.1-.9 0-1.7.4-2.4 1-.7.7-.8 1.1-.8 2.3v8.9h-2.2v-11c0-1.9-.4-2.8-.5-3l2.2-.6c.1.2.5 1.3.5 2.3 1-1.5 2.5-2.4 4-2.4.3 0 .7.1.8.1l-1 2.5zM145.638 50.9V33.4c0-1.7-.3-2.7-.4-2.9l2.2-.4c.1.2.4 1.4.4 3.4V51h-2.2v-.1zm8.9 0-6.5-8 5.4-6.2h2.8l-5.6 6.2 6.9 8h-3zM161.438 44.1v.3c0 1.3.199 2.4.699 3.1.8 1.2 2.1 1.7 3.5 1.7s2.5-.4 3.5-1.3l.901 1.4c-1.3 1.1-3 1.8-4.8 1.8-3.9 0-6.3-2.9-6.3-7.6 0-2.4.499-3.9 1.699-5.4 1.1-1.3 2.5-2 4.301-2 1.5 0 2.9.5 3.9 1.5 1.2 1.3 1.6 2.6 1.6 6v.3h-9v.2zm5.9-4.8c-.5-.8-1.5-1.3-2.6-1.3-2 0-3.201 1.5-3.3 4.4h6.5c0-1.5-.2-2.4-.6-3.1zM179.938 38.3h-3v9.2c0 1.6.4 2.1 1.8 2.1.6 0 1-.1 1.4-.3l.3 1.4c-.7.4-1.5.6-2.5.6-.7 0-1.3-.1-1.9-.4-1-.5-1.4-1.4-1.4-2.8v-9.8h-1.9v-1.7h1.9c0-1.2.2-3.1.2-3.6l2.3-.5c-.1.5-.3 2.4-.3 4.1h3.6l-.5 1.7z",
            fill: "#000"
          }
        )
      ]
    }
  );
};
var LogoVertical = ({ label, width, height }) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "svg",
    {
      "aria-label": label,
      width,
      height,
      role: "img",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 125 91",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M77.334 2.1h-34.5c-1.7 0-3.1 1.4-3.1 3.1v21.2c2.7-2.1 6.1-4.9 12.2-3.4 5.3 1.3 6.4 5.4 14.8 10 5.1 2.8 9.9.6 13.8-1.6V5.3c0-1.8-1.4-3.2-3.2-3.2z",
            fill: "#1A833B"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M69.434 42.4c-1.7-.2-3.1-.8-4.4-1.7-2.9-1.8-5.7-4.7-11.7-7.8-4.7-2.3-9.4-.5-13.6 1.9v23.7c0 1.7 1.4 3.1 3.1 3.1h34.5c1.7 0 3.1-1.4 3.1-3.1V39.2c-3.6 2.1-7.1 3.8-11 3.2z",
            fill: "#1A589F"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M66.734 33c-8.3-4.6-9.5-8.7-14.8-10-6.1-1.5-9.5 1.4-12.2 3.4V34.8c4.2-2.4 8.9-4.3 13.6-1.9 6 3.1 8.8 6 11.7 7.8 1.3.8 2.7 1.4 4.4 1.7 3.9.6 7.4-1.1 11.1-3.2V31.3c-3.9 2.3-8.7 4.5-13.8 1.7z",
            fill: "#fff"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          "path",
          {
            d: "M2.836 87.5V68.4h2.4v19.1h-2.4zm5.1-10.1 7.6 10.1h-2.9l-7.3-10.1 7.4-9h2.9l-7.7 9zM27.236 88c-.9-.3-1.6-1.1-1.9-2-1.3 1.3-2.4 1.9-4 1.9-3.4 0-4.5-2.1-4.5-4 0-3.1 2.7-5 7.4-5h1v-1c0-1.2-.1-1.7-.4-2.2-.4-.5-1-.8-1.9-.8-1.6 0-3.5.8-4.4 1.6l-1.1-1.6c1.8-1.2 3.8-1.9 5.8-1.9 1.9 0 3.3.8 3.8 2.2.3.8.3 1.8.3 3.1l-.1 4.8c-.1 2.3.1 2.8 1.1 3.4l-1.1 1.5zm-3.2-7.6c-3.5 0-4.8 1.2-4.8 3.3 0 1.6.8 2.5 2.4 2.5 1.4 0 2.7-.9 3.4-2.2l.1-3.6h-1.1zM38.435 75.7c-.1-.1-.4-.1-.6-.1-.8 0-1.7.4-2.3 1-.6.6-.8 1.1-.8 2.2v8.7h-2.2V76.8c0-1.8-.4-2.8-.5-3l2.2-.6c.1.2.5 1.2.4 2.3 1-1.4 2.5-2.3 3.9-2.3.3 0 .7.1.8.1l-.9 2.4zM47.034 75.3h-2.9v8.9c0 1.5.4 2.1 1.8 2.1.6 0 1-.1 1.4-.3l.3 1.4c-.7.4-1.5.6-2.5.6-.7 0-1.3-.1-1.8-.4-1-.5-1.3-1.4-1.3-2.8v-9.5h-1.8v-1.6h1.8c0-1.2.2-3.1.2-3.5l2.3-.5c-.1.4-.3 2.4-.3 4h3.5l-.7 1.6zM55.737 87.6h-2.1l-5-13.9 2.2-.5 3.2 9.6c.3 1 .6 2.1.7 2.4h.1c.1-.2.3-1.2.7-2.3l3.2-9.2h2.3l-5.3 13.9zM65.436 80.9v.3c0 1.3.2 2.3.6 3 .8 1.2 2.1 1.7 3.5 1.7 1.3 0 2.4-.4 3.4-1.3l.8 1.4c-1.2 1.1-2.9 1.7-4.7 1.7-3.8 0-6.2-2.8-6.2-7.4 0-2.3.5-3.8 1.7-5.2 1.1-1.3 2.5-1.9 4.1-1.9 1.5 0 2.8.5 3.8 1.5 1.2 1.2 1.6 2.5 1.6 5.8v.3h-8.6v.1zm5.7-4.7c-.5-.8-1.4-1.2-2.5-1.2-2 0-3.1 1.4-3.2 4.3h6.3c0-1.5-.2-2.4-.6-3.1zM84.134 75.7c-.1-.1-.4-.1-.6-.1-.8 0-1.7.4-2.3 1-.6.6-.8 1.1-.8 2.2v8.7h-2.2V76.8c0-1.8-.4-2.8-.5-3l2.2-.6c.1.2.5 1.2.4 2.3 1-1.4 2.5-2.3 3.9-2.3.3 0 .7.1.8.1l-.9 2.4zM87.338 87.5V70.4c0-1.6-.3-2.6-.4-2.8l2.2-.4c.1.2.4 1.4.4 3.3v17.1h-2.2v-.1zm8.6 0-6.4-7.9 5.3-6h2.7l-5.5 6 6.7 7.9h-2.8zM102.737 80.9v.3c0 1.3.2 2.3.6 3 .8 1.2 2.1 1.7 3.5 1.7 1.3 0 2.4-.4 3.4-1.3l.8 1.4c-1.2 1.1-2.9 1.7-4.7 1.7-3.8 0-6.2-2.8-6.2-7.4 0-2.3.5-3.8 1.7-5.2 1.1-1.3 2.5-1.9 4.1-1.9 1.5 0 2.8.5 3.8 1.5 1.2 1.2 1.6 2.5 1.6 5.8v.3h-8.6v.1zm5.7-4.7c-.5-.8-1.4-1.2-2.5-1.2-2 0-3.1 1.4-3.2 4.3h6.3c0-1.5-.2-2.4-.6-3.1zM120.636 75.3h-2.9v8.9c0 1.5.4 2.1 1.8 2.1.6 0 1-.1 1.4-.3l.3 1.4c-.7.4-1.5.6-2.5.6-.7 0-1.3-.1-1.8-.4-1-.5-1.3-1.4-1.3-2.8v-9.5h-1.8v-1.6h1.8c0-1.2.2-3.1.2-3.5l2.3-.5c-.1.4-.3 2.4-.3 4h3.5l-.7 1.6z",
            fill: "#222"
          }
        )
      ]
    }
  );
};
var Logo = ({ variant = "vertical", size, label }) => {
  const { width, height } = computeDimensions(variant, size);
  if (variant === "horizontal") {
    return /* @__PURE__ */ jsxRuntime.jsx(LogoHorizontal, { label, width, height });
  }
  return /* @__PURE__ */ jsxRuntime.jsx(LogoVertical, { label, width, height });
};
var computeDimensions = (variant, size) => {
  const defaultWidth = variant === "horizontal" ? 184 : 125;
  const aspectRatio = variant === "horizontal" ? 64 / 184 : 91 / 125;
  const width = size != null ? size : defaultWidth;
  const height = width * aspectRatio;
  return { width, height };
};
function genericForwardRef(render) {
  return react.forwardRef(render);
}
var SearchAsyncNoRef = ({
  loadOptions,
  onChange,
  placeholder,
  debounceTime,
  autoFocus,
  className,
  isClearable = true,
  dropdownIndicator,
  size,
  defaultOptions,
  variant,
  id,
  isMulti = false,
  noOptionsMessage,
  isDisabled,
  focusBorderColor,
  value,
  optionLabelFormatter
}, ref) => {
  const noOptionsMessageDefault = ({ inputValue }) => {
    if (inputValue.replaceAll(/\s/g, "").length < 1) {
      return null;
    }
    return /* @__PURE__ */ jsxRuntime.jsx(react.Text, { children: "Fant ingen resultater" });
  };
  const loadOptionsDebounce = useDebounce(loadOptions, debounceTime);
  const onChangeWrapper = (newValue, actionMeta) => {
    if (isMulti) {
      const multiChange = onChange;
      multiChange(newValue, actionMeta);
    } else {
      const singleChange = onChange;
      singleChange(newValue);
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsx(
    chakraReactSelect.AsyncSelect,
    {
      components: __spreadValues({
        DropdownIndicator: () => dropdownIndicator != null ? dropdownIndicator : null
      }, !dropdownIndicator ? { IndicatorSeparator: () => null } : {}),
      formatOptionLabel: optionLabelFormatter,
      isClearable,
      autoFocus,
      className: className ? className : "",
      onChange: onChangeWrapper,
      noOptionsMessage: noOptionsMessage || noOptionsMessageDefault,
      loadingMessage: () => /* @__PURE__ */ jsxRuntime.jsx(react.Text, { children: "Laster..." }),
      loadOptions: debounceTime ? loadOptionsDebounce : loadOptions,
      blurInputOnSelect: false,
      placeholder: placeholder ? placeholder : "S\xF8k her...",
      size,
      defaultOptions,
      variant,
      id,
      isMulti,
      isDisabled,
      focusBorderColor,
      value,
      ref
    }
  );
};
var SearchAsync = genericForwardRef(SearchAsyncNoRef);
var useDebounce = (func, delay = 300) => {
  const timer = react$1.useRef();
  react$1.useEffect(() => {
    return () => {
      if (!timer.current)
        return;
      clearTimeout(timer.current);
    };
  }, []);
  return (inputValue, callback) => {
    const newTimer = setTimeout(() => {
      return func(inputValue, callback);
    }, delay);
    clearTimeout(timer.current);
    timer.current = newTimer;
  };
};
var FooterToggleableFlex = (props) => {
  if (props.hidden)
    return null;
  return /* @__PURE__ */ jsxRuntime.jsx(react.Flex, __spreadProps(__spreadValues({ flexDir: props.flexDir || "column", gap: 1, flex: 1 }, props), { children: props.children }));
};
var Footer = ({
  accessibilityUrl,
  excludeContactInfo,
  excludeOpeningHours,
  excludeHelp,
  excludeNews,
  excludeSocialMedia,
  contactInfoEmailAddress = "post@kartverket.no"
}) => {
  const onlyOneIncluded = (!excludeContactInfo || !excludeOpeningHours ? 1 : 0) + (!excludeHelp ? 1 : 0) + (!excludeNews ? 1 : 0) <= 1;
  return /* @__PURE__ */ jsxRuntime.jsx(react.Box, { marginTop: "auto", alignSelf: "flex-end", w: "full", py: "12", px: "6", bg: "#f5f2f2", borderTop: "8px solid #1a833b", children: /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { flexDir: "column", gap: 8, maxWidth: 1200, mx: "auto", children: [
    /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { alignItems: "flex-start", justifyContent: "space-between", gap: 10, flexWrap: "wrap", children: [
      /* @__PURE__ */ jsxRuntime.jsx(Logo, { size: 200, variant: "horizontal" }),
      /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { gap: 8, flex: "1", flexWrap: "wrap", children: [
        /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { gap: 8, flexDir: onlyOneIncluded ? "row" : "column", justify: "space-between", children: [
          /* @__PURE__ */ jsxRuntime.jsxs(FooterToggleableFlex, { hidden: excludeContactInfo, children: [
            /* @__PURE__ */ jsxRuntime.jsx(react.Heading, { size: "md", pb: 2, children: "Kontakt oss" }),
            /* @__PURE__ */ jsxRuntime.jsxs(react.Text, { children: [
              "Telefon:",
              " ",
              /* @__PURE__ */ jsxRuntime.jsxs(Link, { fontWeight: "bold", "aria-label": "Ring Kartverket p\xE5 telefon", href: "tel:32118000", children: [
                " ",
                "32 11 80 00",
                " "
              ] })
            ] }),
            /* @__PURE__ */ jsxRuntime.jsxs(react.Text, { children: [
              "E-post:",
              " ",
              /* @__PURE__ */ jsxRuntime.jsxs(
                Link,
                {
                  fontWeight: "bold",
                  "aria-label": "Send e-post til Kartverket",
                  href: "mailto:" + contactInfoEmailAddress,
                  children: [
                    " ",
                    contactInfoEmailAddress,
                    " "
                  ]
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntime.jsxs(
              Link,
              {
                p: 0,
                fontWeight: "bold",
                "aria-label": "Bes\xF8k Kartverket sin kontaktinformasjon",
                href: "https://kartverket.no/om-kartverket/kontakt-oss",
                children: [
                  " ",
                  "Kontaktinfo og adresser",
                  " "
                ]
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntime.jsxs(FooterToggleableFlex, { hidden: excludeOpeningHours, children: [
            /* @__PURE__ */ jsxRuntime.jsx(react.Text, { fontWeight: "bold", children: " \xC5pningstider (mandag-fredag) " }),
            /* @__PURE__ */ jsxRuntime.jsx(react.Text, { children: " Kundesenter/sentralbord: 09.00-15.00 " }),
            /* @__PURE__ */ jsxRuntime.jsx(react.Text, { children: " Resepsjon: 08.00-15.00 " })
          ] })
        ] }),
        /* @__PURE__ */ jsxRuntime.jsxs(FooterToggleableFlex, { hidden: excludeHelp, children: [
          /* @__PURE__ */ jsxRuntime.jsx(react.Heading, { size: "md", children: " Hjelp " }),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://status.kartverket.no/",
              "aria-label": "Bes\xF8k Kartverket sine driftsmeldinger",
              fontWeight: "bold",
              children: [
                " ",
                "Driftsmeldinger",
                " "
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/om-kartverket/faq",
              "aria-label": "Bes\xF8k Kartverket sin oversikt over ofte spurte sp\xF8rsm\xE5l",
              fontWeight: "bold",
              children: [
                " ",
                "Ofte stilte sp\xF8rsm\xE5l",
                " "
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/eiendom/skjema/alleskjema",
              "aria-label": "Bes\xF8k Kartverket sin oversikt over skjemaer",
              fontWeight: "bold",
              children: [
                " ",
                "Skjemaer",
                " "
              ]
            }
          )
        ] }),
        /* @__PURE__ */ jsxRuntime.jsxs(FooterToggleableFlex, { hidden: excludeNews, children: [
          /* @__PURE__ */ jsxRuntime.jsx(react.Heading, { size: "md", children: " Aktuelt " }),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/om-kartverket/arrangement",
              "aria-label": "Bes\xF8k Kartverket sin oversikt over Arrangementer og kurs",
              fontWeight: "bold",
              children: [
                " ",
                "Arrangementer og kurs",
                " "
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/om-kartverket/nyheter",
              "aria-label": "Bes\xF8k Kartverket sine Nyheter",
              fontWeight: "bold",
              children: [
                " ",
                "Nyheter",
                " "
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/om-kartverket/jobb-i-kartverket",
              "aria-label": "Bes\xF8k Kartverket sin side med informasjon for jobb i Kartverket",
              fontWeight: "bold",
              children: [
                " ",
                "Ledige stillinger",
                " "
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntime.jsxs(
            Link,
            {
              href: "https://kartverket.no/om-kartverket/presse",
              "aria-label": "Bes\xF8k Kartverket sin presse-side",
              fontWeight: "bold",
              children: [
                " ",
                "Presse",
                " "
              ]
            }
          )
        ] })
      ] })
    ] }),
    /* @__PURE__ */ jsxRuntime.jsx(react.Divider, { opacity: "1", borderColor: "gray.400" }),
    /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { align: "center", justify: "space-between", gap: 10, flexWrap: "wrap", children: [
      excludeSocialMedia ? /* @__PURE__ */ jsxRuntime.jsx(react.Box, {}) : /* @__PURE__ */ jsxRuntime.jsxs(FooterToggleableFlex, { align: "center", gap: 3, flexWrap: "wrap", flexDir: "row", children: [
        /* @__PURE__ */ jsxRuntime.jsx(react.Heading, { size: "md", children: " F\xF8lg oss: " }),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://www.facebook.com/kartverket.no",
            "aria-label": "Bes\xF8k Kartverket sin Facebook-side",
            fontWeight: "bold",
            children: "Facebook"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://twitter.com/Kartverket",
            "aria-label": "Bes\xF8k Kartverket sin Twitter-side",
            fontWeight: "bold",
            children: "Twitter"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://www.instagram.com/kartverket.no/",
            "aria-label": "Bes\xF8k Kartverket sin Instagram-side",
            fontWeight: "bold",
            children: "Instagram"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://www.linkedin.com/company/65713",
            "aria-label": "Bes\xF8k Kartverket sin LinkedIn-side",
            fontWeight: "bold",
            children: "LinkedIn"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://pub.dialogapi.no/s/3403d74a-8e34-4ab5-a221-1bf58c425982",
            "aria-label": "Bes\xF8k Kartverket sin side for \xE5 melde seg p\xE5 nyhetsvarsler",
            fontWeight: "bold",
            children: "Newsletter"
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://vimeo.com/kartverket/videos",
            "aria-label": "Bes\xF8k Kartverket sin Video-side",
            fontWeight: "bold",
            children: "Video"
          }
        )
      ] }),
      /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { align: "center", gap: 3, flexWrap: "wrap", children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          Link,
          {
            href: "https://kartverket.no/om-kartverket/personvern",
            "aria-label": "Bes\xF8k Kartverket sin personvernserkl\xE6ring",
            fontWeight: "bold",
            children: "Personvern"
          }
        ),
        accessibilityUrl && /* @__PURE__ */ jsxRuntime.jsxs(Link, { href: accessibilityUrl, "aria-label": "Bes\xF8k denne sidens tilgjengelighetserkl\xE6ring", fontWeight: "bold", children: [
          " ",
          "Tilgjengelighetserkl\xE6ring"
        ] })
      ] })
    ] })
  ] }) });
};
var Header = (props) => {
  const {
    justifyContent = "space-between",
    logoLink = "/",
    logoAltText,
    children,
    showMenuButton = false,
    dropdownMenuChildren,
    showChildrenInMenu = true,
    collapseBreakpoint = "sm",
    onMenuButtonClick,
    gap = 90,
    logoVariant = "horizontal"
  } = props;
  const [isCollapse] = react.useMediaQuery(`(max-width: ${defaultKvibTheme.breakpoints[collapseBreakpoint]})`);
  const [isSm] = react.useMediaQuery(`(max-width: ${defaultKvibTheme.breakpoints["sm"]})`);
  const logoHorizontalSize = isSm ? 110 : 150;
  const logoVerticalSize = isSm ? 70 : 100;
  const headerSize = isSm ? 70 : 90;
  const justify = justifyContent && isCollapse ? "space-between" : justifyContent;
  const showChildren = !isCollapse;
  const { isOpen, onToggle } = react.useDisclosure();
  const showMenuButtonElement = children && (isCollapse || isOpen) || showMenuButton;
  const handleClick = onMenuButtonClick || onToggle;
  return /* @__PURE__ */ jsxRuntime.jsxs(react.Box, { children: [
    /* @__PURE__ */ jsxRuntime.jsxs(
      react.Flex,
      {
        bg: "white",
        borderBottomWidth: "2px",
        borderBottomColor: "gray.200",
        padding: 30,
        height: headerSize,
        alignItems: "center",
        justifyContent: justify,
        gap,
        children: [
          /* @__PURE__ */ jsxRuntime.jsx(Link, { href: logoLink, isExternal: false, children: /* @__PURE__ */ jsxRuntime.jsx(
            Logo,
            {
              label: logoAltText,
              variant: logoVariant,
              size: logoVariant == "horizontal" ? logoHorizontalSize : logoVerticalSize
            }
          ) }),
          showChildren && children,
          showMenuButtonElement && /* @__PURE__ */ jsxRuntime.jsx(IconButton, { "aria-label": "open menu", icon: isOpen ? "close" : "menu", variant: "ghost", onClick: handleClick })
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(react.Collapse, { in: isOpen, animateOpacity: false, children: /* @__PURE__ */ jsxRuntime.jsxs(react.VStack, { bg: "gray.50", borderBottomWidth: "2px", borderBottomColor: "gray.200", padding: 30, gap: 10, children: [
      showChildrenInMenu && children,
      dropdownMenuChildren
    ] }) })
  ] });
};
var FooterInline = ({
  logoLink,
  accessibilityUrl,
  privacyUrl = "https://kartverket.no/om-kartverket/personvern",
  children
}) => {
  return /* @__PURE__ */ jsxRuntime.jsxs(
    react.Flex,
    {
      borderTop: `1px solid ${colors.gray[300]}`,
      marginTop: "1rem",
      justifyContent: "space-between",
      paddingTop: "10px",
      gap: "2rem",
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(Link, { href: logoLink, isExternal: false, children: /* @__PURE__ */ jsxRuntime.jsx(Logo, { size: 130, variant: "horizontal" }) }),
        /* @__PURE__ */ jsxRuntime.jsxs(react.Flex, { alignItems: "center", gap: "1rem", wrap: "wrap", children: [
          children,
          accessibilityUrl && /* @__PURE__ */ jsxRuntime.jsx(Link, { href: accessibilityUrl, "aria-label": "Bes\xF8k denne sidens tilgjengelighetserkl\xE6ring", fontWeight: "bold", children: "Tilgjengelighetserkl\xE6ring" }),
          /* @__PURE__ */ jsxRuntime.jsx(Link, { href: privacyUrl, "aria-label": "Bes\xF8k Kartverket sin personvernserkl\xE6ring", fontWeight: "bold", children: "Personvern" })
        ] })
      ]
    }
  );
};

// ../../node_modules/@swc/helpers/esm/_check_private_redeclaration.js
function _check_private_redeclaration(obj, privateCollection) {
  if (privateCollection.has(obj)) {
    throw new TypeError("Cannot initialize the same private elements twice on an object");
  }
}

// ../../node_modules/@swc/helpers/esm/_class_private_field_init.js
function _class_private_field_init(obj, privateMap, value) {
  _check_private_redeclaration(obj, privateMap);
  privateMap.set(obj, value);
}

// ../../node_modules/@internationalized/date/dist/import.mjs
function $14e0f24ef4ac5c92$export$c19a80a9721b80f6(a, b) {
  return $14e0f24ef4ac5c92$var$timeToMs(a) - $14e0f24ef4ac5c92$var$timeToMs(b);
}
function $14e0f24ef4ac5c92$var$timeToMs(a) {
  return a.hour * 36e5 + a.minute * 6e4 + a.second * 1e3 + a.millisecond;
}
function $735220c2d4774dd3$export$3e2544e88a25bff8(duration) {
  let inverseDuration = {};
  for (let key in duration)
    if (typeof duration[key] === "number")
      inverseDuration[key] = -duration[key];
  return inverseDuration;
}
function $735220c2d4774dd3$export$e5d5e1c1822b6e56(value, fields) {
  let mutableValue = value.copy();
  if (fields.hour != null)
    mutableValue.hour = fields.hour;
  if (fields.minute != null)
    mutableValue.minute = fields.minute;
  if (fields.second != null)
    mutableValue.second = fields.second;
  if (fields.millisecond != null)
    mutableValue.millisecond = fields.millisecond;
  $735220c2d4774dd3$export$7555de1e070510cb(mutableValue);
  return mutableValue;
}
function $735220c2d4774dd3$var$balanceTime(time) {
  time.second += Math.floor(time.millisecond / 1e3);
  time.millisecond = $735220c2d4774dd3$var$nonNegativeMod(time.millisecond, 1e3);
  time.minute += Math.floor(time.second / 60);
  time.second = $735220c2d4774dd3$var$nonNegativeMod(time.second, 60);
  time.hour += Math.floor(time.minute / 60);
  time.minute = $735220c2d4774dd3$var$nonNegativeMod(time.minute, 60);
  let days = Math.floor(time.hour / 24);
  time.hour = $735220c2d4774dd3$var$nonNegativeMod(time.hour, 24);
  return days;
}
function $735220c2d4774dd3$export$7555de1e070510cb(time) {
  time.millisecond = Math.max(0, Math.min(time.millisecond, 1e3));
  time.second = Math.max(0, Math.min(time.second, 59));
  time.minute = Math.max(0, Math.min(time.minute, 59));
  time.hour = Math.max(0, Math.min(time.hour, 23));
}
function $735220c2d4774dd3$var$nonNegativeMod(a, b) {
  let result = a % b;
  if (result < 0)
    result += b;
  return result;
}
function $735220c2d4774dd3$var$addTimeFields(time, duration) {
  time.hour += duration.hours || 0;
  time.minute += duration.minutes || 0;
  time.second += duration.seconds || 0;
  time.millisecond += duration.milliseconds || 0;
  return $735220c2d4774dd3$var$balanceTime(time);
}
function $735220c2d4774dd3$export$7ed87b6bc2506470(time, duration) {
  let res = time.copy();
  $735220c2d4774dd3$var$addTimeFields(res, duration);
  return res;
}
function $735220c2d4774dd3$export$fe34d3a381cd7501(time, duration) {
  return $735220c2d4774dd3$export$7ed87b6bc2506470(time, $735220c2d4774dd3$export$3e2544e88a25bff8(duration));
}
function $735220c2d4774dd3$export$dd02b3e0007dfe28(value, field, amount, options) {
  let mutable = value.copy();
  switch (field) {
    case "hour": {
      let hours = value.hour;
      let min = 0;
      let max = 23;
      if ((options === null || options === void 0 ? void 0 : options.hourCycle) === 12) {
        let isPM = hours >= 12;
        min = isPM ? 12 : 0;
        max = isPM ? 23 : 11;
      }
      mutable.hour = $735220c2d4774dd3$var$cycleValue(hours, amount, min, max, options === null || options === void 0 ? void 0 : options.round);
      break;
    }
    case "minute":
      mutable.minute = $735220c2d4774dd3$var$cycleValue(value.minute, amount, 0, 59, options === null || options === void 0 ? void 0 : options.round);
      break;
    case "second":
      mutable.second = $735220c2d4774dd3$var$cycleValue(value.second, amount, 0, 59, options === null || options === void 0 ? void 0 : options.round);
      break;
    case "millisecond":
      mutable.millisecond = $735220c2d4774dd3$var$cycleValue(value.millisecond, amount, 0, 999, options === null || options === void 0 ? void 0 : options.round);
      break;
    default:
      throw new Error("Unsupported field " + field);
  }
  return mutable;
}
function $735220c2d4774dd3$var$cycleValue(value, amount, min, max, round = false) {
  if (round) {
    value += Math.sign(amount);
    if (value < min)
      value = max;
    let div = Math.abs(amount);
    if (amount > 0)
      value = Math.ceil(value / div) * div;
    else
      value = Math.floor(value / div) * div;
    if (value > max)
      value = min;
  } else {
    value += amount;
    if (value < min)
      value = max - (min - value - 1);
    else if (value > max)
      value = min + (value - max - 1);
  }
  return value;
}
var $fae977aafc393c5c$var$TIME_RE = /^(\d{2})(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?$/;
function $fae977aafc393c5c$export$c9698ec7f05a07e1(value) {
  let m = value.match($fae977aafc393c5c$var$TIME_RE);
  if (!m)
    throw new Error("Invalid ISO 8601 time string: " + value);
  return new ($35ea8db9cb2ccb90$export$680ea196effce5f)($fae977aafc393c5c$var$parseNumber(m[1], 0, 23), m[2] ? $fae977aafc393c5c$var$parseNumber(m[2], 0, 59) : 0, m[3] ? $fae977aafc393c5c$var$parseNumber(m[3], 0, 59) : 0, m[4] ? $fae977aafc393c5c$var$parseNumber(m[4], 0, Infinity) * 1e3 : 0);
}
function $fae977aafc393c5c$var$parseNumber(value, min, max) {
  let val = Number(value);
  if (val < min || val > max)
    throw new RangeError(`Value out of range: ${min} <= ${val} <= ${max}`);
  return val;
}
function $fae977aafc393c5c$export$f59dee82248f5ad4(time) {
  return `${String(time.hour).padStart(2, "0")}:${String(time.minute).padStart(2, "0")}:${String(time.second).padStart(2, "0")}${time.millisecond ? String(time.millisecond / 1e3).slice(1) : ""}`;
}
var $35ea8db9cb2ccb90$var$_type1 = /* @__PURE__ */ new WeakMap();
var $35ea8db9cb2ccb90$export$680ea196effce5f = class _$35ea8db9cb2ccb90$export$680ea196effce5f {
  /** Returns a copy of this time. */
  copy() {
    return new _$35ea8db9cb2ccb90$export$680ea196effce5f(this.hour, this.minute, this.second, this.millisecond);
  }
  /** Returns a new `Time` with the given duration added to it. */
  add(duration) {
    return ($735220c2d4774dd3$export$7ed87b6bc2506470)(this, duration);
  }
  /** Returns a new `Time` with the given duration subtracted from it. */
  subtract(duration) {
    return ($735220c2d4774dd3$export$fe34d3a381cd7501)(this, duration);
  }
  /** Returns a new `Time` with the given fields set to the provided values. Other fields will be constrained accordingly. */
  set(fields) {
    return ($735220c2d4774dd3$export$e5d5e1c1822b6e56)(this, fields);
  }
  /**
  * Returns a new `Time` with the given field adjusted by a specified amount.
  * When the resulting value reaches the limits of the field, it wraps around.
  */
  cycle(field, amount, options) {
    return ($735220c2d4774dd3$export$dd02b3e0007dfe28)(this, field, amount, options);
  }
  /** Converts the time to an ISO 8601 formatted string. */
  toString() {
    return ($fae977aafc393c5c$export$f59dee82248f5ad4)(this);
  }
  /** Compares this time with another. A negative result indicates that this time is before the given one, and a positive time indicates that it is after. */
  compare(b) {
    return ($14e0f24ef4ac5c92$export$c19a80a9721b80f6)(this, b);
  }
  constructor(hour = 0, minute = 0, second = 0, millisecond = 0) {
    (_class_private_field_init)(this, $35ea8db9cb2ccb90$var$_type1, {
      writable: true,
      value: void 0
    });
    this.hour = hour;
    this.minute = minute;
    this.second = second;
    this.millisecond = millisecond;
    ($735220c2d4774dd3$export$7555de1e070510cb)(this);
  }
};
var DateTimeSegment = (_a) => {
  var _b = _a, { segment, state: state2, colorScheme } = _b, props = __objRest(_b, ["segment", "state", "colorScheme"]);
  const ref = react$1.useRef(null);
  const { segmentProps } = reactAria.useDateSegment(segment, state2, ref);
  const styles = react.useStyleConfig("Timepicker", props);
  return /* @__PURE__ */ jsxRuntime.jsx(react.Box, __spreadProps(__spreadValues({}, segmentProps), { ref, paddingX: "1px", outline: "none", borderRadius: "sm", __css: __spreadValues({}, styles), children: isPaddable(segment.type) ? segment.text.padStart(2, "0") : segment.text }));
};
var isPaddable = (segmentType) => segmentType === "month" || segmentType === "day" || segmentType === "hour" || segmentType === "minute" || segmentType === "second";
var TimeField = (_a) => {
  var _b = _a, { state: state2, colorScheme } = _b, props = __objRest(_b, ["state", "colorScheme"]);
  const ref = react$1.useRef(null);
  const { fieldProps } = reactAria.useTimeField(props, state2, ref);
  return /* @__PURE__ */ jsxRuntime.jsxs(react.Box, { children: [
    /* @__PURE__ */ jsxRuntime.jsx(react.Flex, __spreadProps(__spreadValues({}, fieldProps), { ref, children: state2.segments.map((segment) => /* @__PURE__ */ jsxRuntime.jsx(DateTimeSegment, { segment, state: state2, colorScheme }, segment.type)) })),
    /* @__PURE__ */ jsxRuntime.jsx("input", { type: "hidden", value: getTimestampFromTime(state2.value), name: props.name })
  ] });
};
var Timepicker = ({
  size,
  width = "fit-content",
  variant = "outline",
  colorScheme,
  value,
  defaultValue = getCurrentTime(),
  onChange = () => {
  },
  isDisabled: isDisabledExternally = false,
  isInvalid: isInvalidExternally = false,
  minuteInterval = 30,
  ariaLabel
}) => {
  const formControlContext = react.useFormControlContext();
  const isDisabledFromForm = (formControlContext == null ? void 0 : formControlContext.isDisabled) || false;
  const isInvalidFromForm = (formControlContext == null ? void 0 : formControlContext.isInvalid) || false;
  const isDisabled = isDisabledExternally || isDisabledFromForm;
  const isInvalid = isInvalidExternally || isInvalidFromForm;
  const state2 = reactStately.useTimeFieldState({
    value,
    defaultValue,
    onChange,
    locale: "nb",
    isDisabled,
    isInvalid
  });
  const dateTime = state2.value;
  const buttonSize = size === "lg" ? "sm" : "xs";
  const inputRef = react$1.useRef(null);
  const [isFocused, setIsFocused] = react$1.useState(false);
  const adjustTime = (direction) => {
    if (!dateTime) {
      return;
    }
    let minutesAdjustment;
    if (direction === "backward") {
      minutesAdjustment = -(dateTime.minute % minuteInterval || minuteInterval);
    } else {
      minutesAdjustment = minuteInterval - dateTime.minute % minuteInterval || minuteInterval;
    }
    state2.setValue(
      state2.value.add({
        minutes: minutesAdjustment
      })
    );
  };
  const focusStyles = getFocusStyles(isFocused, isInvalid, variant);
  return /* @__PURE__ */ jsxRuntime.jsxs(
    react.Input,
    {
      as: "div",
      "aria-label": ariaLabel || "timepicker",
      display: "flex",
      variant,
      size,
      width,
      paddingX: 2,
      alignItems: "center",
      justifyContent: "space-between",
      gap: 2,
      isDisabled,
      isInvalid,
      onClick: () => {
        var _a;
        return (_a = inputRef.current) == null ? void 0 : _a.focus();
      },
      onFocus: () => setIsFocused(true),
      onBlur: () => setIsFocused(false),
      ref: inputRef,
      sx: focusStyles,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(
          IconButton,
          {
            onClick: () => adjustTime("backward"),
            size: buttonSize,
            colorScheme,
            variant: "ghost",
            icon: "chevron_left",
            "aria-label": "earlier time",
            isDisabled
          }
        ),
        /* @__PURE__ */ jsxRuntime.jsx(TimeField, { state: state2, colorScheme }),
        /* @__PURE__ */ jsxRuntime.jsx(
          IconButton,
          {
            onClick: () => adjustTime("forward"),
            size: buttonSize,
            colorScheme,
            variant: "ghost",
            icon: "chevron_right",
            "aria-label": "later time",
            isDisabled
          }
        )
      ]
    }
  );
};
var getCurrentTime = () => $fae977aafc393c5c$export$c9698ec7f05a07e1((/* @__PURE__ */ new Date()).toTimeString().split(" ")[0]);
var getTimestampFromTime = (time) => {
  var _a, _b;
  return `${(_a = time == null ? void 0 : time.hour) != null ? _a : 0}:${(_b = time == null ? void 0 : time.minute) != null ? _b : 0}`;
};
var getFocusStyles = (isFocused, isInvalid, variant) => {
  if (isFocused) {
    switch (variant) {
      case "outline":
        return {
          borderColor: "blue.500",
          boxShadow: `0 0 0 1px ${defaultKvibTheme.colors.blue[500]}`,
          _hover: { borderColor: isInvalid != null ? isInvalid : "blue.500" }
        };
      case "flushed":
        return {
          borderColor: "blue.500",
          boxShadow: `0 1px 0 0 ${defaultKvibTheme.colors.blue[500]}`,
          _hover: { borderColor: isInvalid != null ? isInvalid : "blue.500" }
        };
      default:
        return {
          borderColor: "blue.500",
          _hover: { borderColor: isInvalid != null ? isInvalid : "blue.500" }
        };
    }
  } else {
    return {};
  }
};

Object.defineProperty(exports, 'AbsoluteCenter', {
  enumerable: true,
  get: function () { return react.AbsoluteCenter; }
});
Object.defineProperty(exports, 'Accordion', {
  enumerable: true,
  get: function () { return react.Accordion; }
});
Object.defineProperty(exports, 'AccordionButton', {
  enumerable: true,
  get: function () { return react.AccordionButton; }
});
Object.defineProperty(exports, 'AccordionIcon', {
  enumerable: true,
  get: function () { return react.AccordionIcon; }
});
Object.defineProperty(exports, 'AccordionItem', {
  enumerable: true,
  get: function () { return react.AccordionItem; }
});
Object.defineProperty(exports, 'AccordionPanel', {
  enumerable: true,
  get: function () { return react.AccordionPanel; }
});
Object.defineProperty(exports, 'AlertDescription', {
  enumerable: true,
  get: function () { return react.AlertDescription; }
});
Object.defineProperty(exports, 'AlertDialog', {
  enumerable: true,
  get: function () { return react.AlertDialog; }
});
Object.defineProperty(exports, 'AlertDialogBody', {
  enumerable: true,
  get: function () { return react.AlertDialogBody; }
});
Object.defineProperty(exports, 'AlertDialogCloseButton', {
  enumerable: true,
  get: function () { return react.AlertDialogCloseButton; }
});
Object.defineProperty(exports, 'AlertDialogContent', {
  enumerable: true,
  get: function () { return react.AlertDialogContent; }
});
Object.defineProperty(exports, 'AlertDialogFooter', {
  enumerable: true,
  get: function () { return react.AlertDialogFooter; }
});
Object.defineProperty(exports, 'AlertDialogHeader', {
  enumerable: true,
  get: function () { return react.AlertDialogHeader; }
});
Object.defineProperty(exports, 'AlertDialogOverlay', {
  enumerable: true,
  get: function () { return react.AlertDialogOverlay; }
});
Object.defineProperty(exports, 'AlertIcon', {
  enumerable: true,
  get: function () { return react.AlertIcon; }
});
Object.defineProperty(exports, 'AlertTitle', {
  enumerable: true,
  get: function () { return react.AlertTitle; }
});
Object.defineProperty(exports, 'AspectRatio', {
  enumerable: true,
  get: function () { return react.AspectRatio; }
});
Object.defineProperty(exports, 'Avatar', {
  enumerable: true,
  get: function () { return react.Avatar; }
});
Object.defineProperty(exports, 'AvatarBadge', {
  enumerable: true,
  get: function () { return react.AvatarBadge; }
});
Object.defineProperty(exports, 'AvatarGroup', {
  enumerable: true,
  get: function () { return react.AvatarGroup; }
});
Object.defineProperty(exports, 'Box', {
  enumerable: true,
  get: function () { return react.Box; }
});
Object.defineProperty(exports, 'Breadcrumb', {
  enumerable: true,
  get: function () { return react.Breadcrumb; }
});
Object.defineProperty(exports, 'BreadcrumbItem', {
  enumerable: true,
  get: function () { return react.BreadcrumbItem; }
});
Object.defineProperty(exports, 'BreadcrumbLink', {
  enumerable: true,
  get: function () { return react.BreadcrumbLink; }
});
Object.defineProperty(exports, 'BreadcrumbSeparator', {
  enumerable: true,
  get: function () { return react.BreadcrumbSeparator; }
});
Object.defineProperty(exports, 'CardBody', {
  enumerable: true,
  get: function () { return react.CardBody; }
});
Object.defineProperty(exports, 'CardFooter', {
  enumerable: true,
  get: function () { return react.CardFooter; }
});
Object.defineProperty(exports, 'CardHeader', {
  enumerable: true,
  get: function () { return react.CardHeader; }
});
Object.defineProperty(exports, 'Center', {
  enumerable: true,
  get: function () { return react.Center; }
});
Object.defineProperty(exports, 'Checkbox', {
  enumerable: true,
  get: function () { return react.Checkbox; }
});
Object.defineProperty(exports, 'CheckboxGroup', {
  enumerable: true,
  get: function () { return react.CheckboxGroup; }
});
Object.defineProperty(exports, 'Circle', {
  enumerable: true,
  get: function () { return react.Circle; }
});
Object.defineProperty(exports, 'CircularProgress', {
  enumerable: true,
  get: function () { return react.CircularProgress; }
});
Object.defineProperty(exports, 'CircularProgressLabel', {
  enumerable: true,
  get: function () { return react.CircularProgressLabel; }
});
Object.defineProperty(exports, 'CloseButton', {
  enumerable: true,
  get: function () { return react.CloseButton; }
});
Object.defineProperty(exports, 'Code', {
  enumerable: true,
  get: function () { return react.Code; }
});
Object.defineProperty(exports, 'Collapse', {
  enumerable: true,
  get: function () { return react.Collapse; }
});
Object.defineProperty(exports, 'Container', {
  enumerable: true,
  get: function () { return react.Container; }
});
Object.defineProperty(exports, 'Divider', {
  enumerable: true,
  get: function () { return react.Divider; }
});
Object.defineProperty(exports, 'Drawer', {
  enumerable: true,
  get: function () { return react.Drawer; }
});
Object.defineProperty(exports, 'DrawerBody', {
  enumerable: true,
  get: function () { return react.DrawerBody; }
});
Object.defineProperty(exports, 'DrawerCloseButton', {
  enumerable: true,
  get: function () { return react.DrawerCloseButton; }
});
Object.defineProperty(exports, 'DrawerContent', {
  enumerable: true,
  get: function () { return react.DrawerContent; }
});
Object.defineProperty(exports, 'DrawerFooter', {
  enumerable: true,
  get: function () { return react.DrawerFooter; }
});
Object.defineProperty(exports, 'DrawerHeader', {
  enumerable: true,
  get: function () { return react.DrawerHeader; }
});
Object.defineProperty(exports, 'DrawerOverlay', {
  enumerable: true,
  get: function () { return react.DrawerOverlay; }
});
Object.defineProperty(exports, 'Editable', {
  enumerable: true,
  get: function () { return react.Editable; }
});
Object.defineProperty(exports, 'EditableInput', {
  enumerable: true,
  get: function () { return react.EditableInput; }
});
Object.defineProperty(exports, 'EditablePreview', {
  enumerable: true,
  get: function () { return react.EditablePreview; }
});
Object.defineProperty(exports, 'EditableTextarea', {
  enumerable: true,
  get: function () { return react.EditableTextarea; }
});
Object.defineProperty(exports, 'Fade', {
  enumerable: true,
  get: function () { return react.Fade; }
});
Object.defineProperty(exports, 'Flex', {
  enumerable: true,
  get: function () { return react.Flex; }
});
Object.defineProperty(exports, 'Grid', {
  enumerable: true,
  get: function () { return react.Grid; }
});
Object.defineProperty(exports, 'GridItem', {
  enumerable: true,
  get: function () { return react.GridItem; }
});
Object.defineProperty(exports, 'HStack', {
  enumerable: true,
  get: function () { return react.HStack; }
});
Object.defineProperty(exports, 'Heading', {
  enumerable: true,
  get: function () { return react.Heading; }
});
Object.defineProperty(exports, 'Hide', {
  enumerable: true,
  get: function () { return react.Hide; }
});
Object.defineProperty(exports, 'Highlight', {
  enumerable: true,
  get: function () { return react.Highlight; }
});
Object.defineProperty(exports, 'Image', {
  enumerable: true,
  get: function () { return react.Image; }
});
Object.defineProperty(exports, 'Input', {
  enumerable: true,
  get: function () { return react.Input; }
});
Object.defineProperty(exports, 'InputAddon', {
  enumerable: true,
  get: function () { return react.InputAddon; }
});
Object.defineProperty(exports, 'InputGroup', {
  enumerable: true,
  get: function () { return react.InputGroup; }
});
Object.defineProperty(exports, 'InputLeftAddon', {
  enumerable: true,
  get: function () { return react.InputLeftAddon; }
});
Object.defineProperty(exports, 'InputLeftElement', {
  enumerable: true,
  get: function () { return react.InputLeftElement; }
});
Object.defineProperty(exports, 'InputRightAddon', {
  enumerable: true,
  get: function () { return react.InputRightAddon; }
});
Object.defineProperty(exports, 'InputRightElement', {
  enumerable: true,
  get: function () { return react.InputRightElement; }
});
Object.defineProperty(exports, 'Kbd', {
  enumerable: true,
  get: function () { return react.Kbd; }
});
Object.defineProperty(exports, 'LinkBox', {
  enumerable: true,
  get: function () { return react.LinkBox; }
});
Object.defineProperty(exports, 'LinkOverlay', {
  enumerable: true,
  get: function () { return react.LinkOverlay; }
});
Object.defineProperty(exports, 'List', {
  enumerable: true,
  get: function () { return react.List; }
});
Object.defineProperty(exports, 'ListIcon', {
  enumerable: true,
  get: function () { return react.ListIcon; }
});
Object.defineProperty(exports, 'ListItem', {
  enumerable: true,
  get: function () { return react.ListItem; }
});
Object.defineProperty(exports, 'Menu', {
  enumerable: true,
  get: function () { return react.Menu; }
});
Object.defineProperty(exports, 'MenuButton', {
  enumerable: true,
  get: function () { return react.MenuButton; }
});
Object.defineProperty(exports, 'MenuDivider', {
  enumerable: true,
  get: function () { return react.MenuDivider; }
});
Object.defineProperty(exports, 'MenuGroup', {
  enumerable: true,
  get: function () { return react.MenuGroup; }
});
Object.defineProperty(exports, 'MenuItem', {
  enumerable: true,
  get: function () { return react.MenuItem; }
});
Object.defineProperty(exports, 'MenuItemOption', {
  enumerable: true,
  get: function () { return react.MenuItemOption; }
});
Object.defineProperty(exports, 'MenuList', {
  enumerable: true,
  get: function () { return react.MenuList; }
});
Object.defineProperty(exports, 'MenuOptionGroup', {
  enumerable: true,
  get: function () { return react.MenuOptionGroup; }
});
Object.defineProperty(exports, 'Modal', {
  enumerable: true,
  get: function () { return react.Modal; }
});
Object.defineProperty(exports, 'ModalBody', {
  enumerable: true,
  get: function () { return react.ModalBody; }
});
Object.defineProperty(exports, 'ModalCloseButton', {
  enumerable: true,
  get: function () { return react.ModalCloseButton; }
});
Object.defineProperty(exports, 'ModalContent', {
  enumerable: true,
  get: function () { return react.ModalContent; }
});
Object.defineProperty(exports, 'ModalFooter', {
  enumerable: true,
  get: function () { return react.ModalFooter; }
});
Object.defineProperty(exports, 'ModalHeader', {
  enumerable: true,
  get: function () { return react.ModalHeader; }
});
Object.defineProperty(exports, 'ModalOverlay', {
  enumerable: true,
  get: function () { return react.ModalOverlay; }
});
Object.defineProperty(exports, 'OrderedList', {
  enumerable: true,
  get: function () { return react.OrderedList; }
});
Object.defineProperty(exports, 'Popover', {
  enumerable: true,
  get: function () { return react.Popover; }
});
Object.defineProperty(exports, 'PopoverAnchor', {
  enumerable: true,
  get: function () { return react.PopoverAnchor; }
});
Object.defineProperty(exports, 'PopoverArrow', {
  enumerable: true,
  get: function () { return react.PopoverArrow; }
});
Object.defineProperty(exports, 'PopoverBody', {
  enumerable: true,
  get: function () { return react.PopoverBody; }
});
Object.defineProperty(exports, 'PopoverCloseButton', {
  enumerable: true,
  get: function () { return react.PopoverCloseButton; }
});
Object.defineProperty(exports, 'PopoverContent', {
  enumerable: true,
  get: function () { return react.PopoverContent; }
});
Object.defineProperty(exports, 'PopoverFooter', {
  enumerable: true,
  get: function () { return react.PopoverFooter; }
});
Object.defineProperty(exports, 'PopoverHeader', {
  enumerable: true,
  get: function () { return react.PopoverHeader; }
});
Object.defineProperty(exports, 'PopoverTrigger', {
  enumerable: true,
  get: function () { return react.PopoverTrigger; }
});
Object.defineProperty(exports, 'Portal', {
  enumerable: true,
  get: function () { return react.Portal; }
});
Object.defineProperty(exports, 'Progress', {
  enumerable: true,
  get: function () { return react.Progress; }
});
Object.defineProperty(exports, 'Radio', {
  enumerable: true,
  get: function () { return react.Radio; }
});
Object.defineProperty(exports, 'RadioGroup', {
  enumerable: true,
  get: function () { return react.RadioGroup; }
});
Object.defineProperty(exports, 'RangeSlider', {
  enumerable: true,
  get: function () { return react.RangeSlider; }
});
Object.defineProperty(exports, 'RangeSliderFilledTrack', {
  enumerable: true,
  get: function () { return react.RangeSliderFilledTrack; }
});
Object.defineProperty(exports, 'RangeSliderMark', {
  enumerable: true,
  get: function () { return react.RangeSliderMark; }
});
Object.defineProperty(exports, 'RangeSliderThumb', {
  enumerable: true,
  get: function () { return react.RangeSliderThumb; }
});
Object.defineProperty(exports, 'RangeSliderTrack', {
  enumerable: true,
  get: function () { return react.RangeSliderTrack; }
});
Object.defineProperty(exports, 'ScaleFade', {
  enumerable: true,
  get: function () { return react.ScaleFade; }
});
Object.defineProperty(exports, 'Select', {
  enumerable: true,
  get: function () { return react.Select; }
});
Object.defineProperty(exports, 'Show', {
  enumerable: true,
  get: function () { return react.Show; }
});
Object.defineProperty(exports, 'SimpleGrid', {
  enumerable: true,
  get: function () { return react.SimpleGrid; }
});
Object.defineProperty(exports, 'Skeleton', {
  enumerable: true,
  get: function () { return react.Skeleton; }
});
Object.defineProperty(exports, 'SkeletonCircle', {
  enumerable: true,
  get: function () { return react.SkeletonCircle; }
});
Object.defineProperty(exports, 'SkeletonText', {
  enumerable: true,
  get: function () { return react.SkeletonText; }
});
Object.defineProperty(exports, 'Slide', {
  enumerable: true,
  get: function () { return react.Slide; }
});
Object.defineProperty(exports, 'SlideFade', {
  enumerable: true,
  get: function () { return react.SlideFade; }
});
Object.defineProperty(exports, 'Slider', {
  enumerable: true,
  get: function () { return react.Slider; }
});
Object.defineProperty(exports, 'SliderFilledTrack', {
  enumerable: true,
  get: function () { return react.SliderFilledTrack; }
});
Object.defineProperty(exports, 'SliderMark', {
  enumerable: true,
  get: function () { return react.SliderMark; }
});
Object.defineProperty(exports, 'SliderThumb', {
  enumerable: true,
  get: function () { return react.SliderThumb; }
});
Object.defineProperty(exports, 'SliderTrack', {
  enumerable: true,
  get: function () { return react.SliderTrack; }
});
Object.defineProperty(exports, 'Spacer', {
  enumerable: true,
  get: function () { return react.Spacer; }
});
Object.defineProperty(exports, 'Spinner', {
  enumerable: true,
  get: function () { return react.Spinner; }
});
Object.defineProperty(exports, 'Square', {
  enumerable: true,
  get: function () { return react.Square; }
});
Object.defineProperty(exports, 'Stack', {
  enumerable: true,
  get: function () { return react.Stack; }
});
Object.defineProperty(exports, 'StackDivider', {
  enumerable: true,
  get: function () { return react.StackDivider; }
});
Object.defineProperty(exports, 'Stat', {
  enumerable: true,
  get: function () { return react.Stat; }
});
Object.defineProperty(exports, 'StatArrow', {
  enumerable: true,
  get: function () { return react.StatArrow; }
});
Object.defineProperty(exports, 'StatGroup', {
  enumerable: true,
  get: function () { return react.StatGroup; }
});
Object.defineProperty(exports, 'StatHelpText', {
  enumerable: true,
  get: function () { return react.StatHelpText; }
});
Object.defineProperty(exports, 'StatLabel', {
  enumerable: true,
  get: function () { return react.StatLabel; }
});
Object.defineProperty(exports, 'StatNumber', {
  enumerable: true,
  get: function () { return react.StatNumber; }
});
Object.defineProperty(exports, 'Step', {
  enumerable: true,
  get: function () { return react.Step; }
});
Object.defineProperty(exports, 'StepDescription', {
  enumerable: true,
  get: function () { return react.StepDescription; }
});
Object.defineProperty(exports, 'StepIcon', {
  enumerable: true,
  get: function () { return react.StepIcon; }
});
Object.defineProperty(exports, 'StepIndicator', {
  enumerable: true,
  get: function () { return react.StepIndicator; }
});
Object.defineProperty(exports, 'StepNumber', {
  enumerable: true,
  get: function () { return react.StepNumber; }
});
Object.defineProperty(exports, 'StepSeparator', {
  enumerable: true,
  get: function () { return react.StepSeparator; }
});
Object.defineProperty(exports, 'StepStatus', {
  enumerable: true,
  get: function () { return react.StepStatus; }
});
Object.defineProperty(exports, 'StepTitle', {
  enumerable: true,
  get: function () { return react.StepTitle; }
});
Object.defineProperty(exports, 'Stepper', {
  enumerable: true,
  get: function () { return react.Stepper; }
});
Object.defineProperty(exports, 'Switch', {
  enumerable: true,
  get: function () { return react.Switch; }
});
Object.defineProperty(exports, 'Tab', {
  enumerable: true,
  get: function () { return react.Tab; }
});
Object.defineProperty(exports, 'TabList', {
  enumerable: true,
  get: function () { return react.TabList; }
});
Object.defineProperty(exports, 'TabPanel', {
  enumerable: true,
  get: function () { return react.TabPanel; }
});
Object.defineProperty(exports, 'TabPanels', {
  enumerable: true,
  get: function () { return react.TabPanels; }
});
Object.defineProperty(exports, 'Table', {
  enumerable: true,
  get: function () { return react.Table; }
});
Object.defineProperty(exports, 'TableCaption', {
  enumerable: true,
  get: function () { return react.TableCaption; }
});
Object.defineProperty(exports, 'TableContainer', {
  enumerable: true,
  get: function () { return react.TableContainer; }
});
Object.defineProperty(exports, 'Tag', {
  enumerable: true,
  get: function () { return react.Tag; }
});
Object.defineProperty(exports, 'TagCloseButton', {
  enumerable: true,
  get: function () { return react.TagCloseButton; }
});
Object.defineProperty(exports, 'TagLabel', {
  enumerable: true,
  get: function () { return react.TagLabel; }
});
Object.defineProperty(exports, 'TagLeftIcon', {
  enumerable: true,
  get: function () { return react.TagLeftIcon; }
});
Object.defineProperty(exports, 'TagRightIcon', {
  enumerable: true,
  get: function () { return react.TagRightIcon; }
});
Object.defineProperty(exports, 'Tbody', {
  enumerable: true,
  get: function () { return react.Tbody; }
});
Object.defineProperty(exports, 'Td', {
  enumerable: true,
  get: function () { return react.Td; }
});
Object.defineProperty(exports, 'Text', {
  enumerable: true,
  get: function () { return react.Text; }
});
Object.defineProperty(exports, 'Textarea', {
  enumerable: true,
  get: function () { return react.Textarea; }
});
Object.defineProperty(exports, 'Tfoot', {
  enumerable: true,
  get: function () { return react.Tfoot; }
});
Object.defineProperty(exports, 'Th', {
  enumerable: true,
  get: function () { return react.Th; }
});
Object.defineProperty(exports, 'Thead', {
  enumerable: true,
  get: function () { return react.Thead; }
});
Object.defineProperty(exports, 'Tooltip', {
  enumerable: true,
  get: function () { return react.Tooltip; }
});
Object.defineProperty(exports, 'Tr', {
  enumerable: true,
  get: function () { return react.Tr; }
});
Object.defineProperty(exports, 'UnorderedList', {
  enumerable: true,
  get: function () { return react.UnorderedList; }
});
Object.defineProperty(exports, 'VStack', {
  enumerable: true,
  get: function () { return react.VStack; }
});
Object.defineProperty(exports, 'VisuallyHidden', {
  enumerable: true,
  get: function () { return react.VisuallyHidden; }
});
Object.defineProperty(exports, 'VisuallyHiddenInput', {
  enumerable: true,
  get: function () { return react.VisuallyHiddenInput; }
});
Object.defineProperty(exports, 'Wrap', {
  enumerable: true,
  get: function () { return react.Wrap; }
});
Object.defineProperty(exports, 'WrapItem', {
  enumerable: true,
  get: function () { return react.WrapItem; }
});
Object.defineProperty(exports, 'extendTheme', {
  enumerable: true,
  get: function () { return react.extendTheme; }
});
Object.defineProperty(exports, 'forwardRef', {
  enumerable: true,
  get: function () { return react.forwardRef; }
});
Object.defineProperty(exports, 'useBoolean', {
  enumerable: true,
  get: function () { return react.useBoolean; }
});
Object.defineProperty(exports, 'useBreakpointValue', {
  enumerable: true,
  get: function () { return react.useBreakpointValue; }
});
Object.defineProperty(exports, 'useClipboard', {
  enumerable: true,
  get: function () { return react.useClipboard; }
});
Object.defineProperty(exports, 'useConst', {
  enumerable: true,
  get: function () { return react.useConst; }
});
Object.defineProperty(exports, 'useControllableProp', {
  enumerable: true,
  get: function () { return react.useControllableProp; }
});
Object.defineProperty(exports, 'useControllableState', {
  enumerable: true,
  get: function () { return react.useControllableState; }
});
Object.defineProperty(exports, 'useDimensions', {
  enumerable: true,
  get: function () { return react.useDimensions; }
});
Object.defineProperty(exports, 'useDisclosure', {
  enumerable: true,
  get: function () { return react.useDisclosure; }
});
Object.defineProperty(exports, 'useMediaQuery', {
  enumerable: true,
  get: function () { return react.useMediaQuery; }
});
Object.defineProperty(exports, 'useMergeRefs', {
  enumerable: true,
  get: function () { return react.useMergeRefs; }
});
Object.defineProperty(exports, 'useOutsideClick', {
  enumerable: true,
  get: function () { return react.useOutsideClick; }
});
Object.defineProperty(exports, 'usePrefersReducedMotion', {
  enumerable: true,
  get: function () { return react.usePrefersReducedMotion; }
});
Object.defineProperty(exports, 'useSteps', {
  enumerable: true,
  get: function () { return react.useSteps; }
});
Object.defineProperty(exports, 'useTheme', {
  enumerable: true,
  get: function () { return react.useTheme; }
});
Object.defineProperty(exports, 'useToast', {
  enumerable: true,
  get: function () { return react.useToast; }
});
Object.defineProperty(exports, 'useToken', {
  enumerable: true,
  get: function () { return react.useToken; }
});
Object.defineProperty(exports, 'withDefaultColorScheme', {
  enumerable: true,
  get: function () { return react.withDefaultColorScheme; }
});
Object.defineProperty(exports, 'SkipNavContent', {
  enumerable: true,
  get: function () { return skipNav.SkipNavContent; }
});
Object.defineProperty(exports, 'SkipNavLink', {
  enumerable: true,
  get: function () { return skipNav.SkipNavLink; }
});
exports.Alert = Alert;
exports.Badge = Badge;
exports.Button = Button;
exports.ButtonGroup = ButtonGroup;
exports.Card = Card;
exports.Datepicker = Datepicker;
exports.FileUpload = FileUpload;
exports.Footer = Footer;
exports.FooterInline = FooterInline;
exports.FormControl = FormControl;
exports.FormErrorMessage = FormErrorMessage;
exports.FormHelperText = FormHelperText;
exports.FormLabel = FormLabel;
exports.Header = Header;
exports.Icon = Icon;
exports.IconButton = IconButton;
exports.KvibProvider = KvibProvider;
exports.Link = Link;
exports.Logo = Logo;
exports.NumberInput = NumberInput;
exports.NumberInputField = NumberInputField;
exports.NumberInputStepper = NumberInputStepper;
exports.PinInput = PinInput;
exports.PinInputField = PinInputField;
exports.Search = Search;
exports.SearchAsync = SearchAsync;
exports.Tabs = Tabs;
exports.Timepicker = Timepicker;
exports.defaultKvibTheme = defaultKvibTheme;
exports.getCurrentTime = getCurrentTime;
exports.getTimestampFromTime = getTimestampFromTime;
